# 专业分类c1c2

In [3]:
import pandas as pd

csv_file = "7.21/major.csv"
c=['id','majorName','year','minScore','avgScore','maxScore','firstlevelIDs','categoryID_id','collegeID_id','provinceID_id']
csv_data = pd.read_csv(csv_file,encoding='GBK',names=c)
csv_data

id           majorName  year  minScore  avgScore  maxScore  \
0             0                 all  2019       615       NaN       NaN   
1            42                 all  2019       645       NaN       NaN   
2            43                 all  2019       711       NaN       NaN   
3            44                 all  2019       646       NaN       NaN   
4            45                 all  2019       683       NaN       NaN   
...         ...                 ...   ...       ...       ...       ...   
186461  5528704  野生动物与自然保护区管理（国家专项）  2020       611       NaN       NaN   
186462  5531969    计算机科学与技术(中外合作办学)  2020       589       NaN       NaN   
186463  5535235          化学(中外合作办学)  2020       577       NaN       NaN   
186464  5538502        生物技术(中外合作办学)  2020       575       NaN       NaN   
186465  5541770        工程管理(中外合作办学)  2020       547       NaN       NaN   

        firstlevelIDs  categoryID_id  collegeID_id  provinceID_id  
0                 NaN              2         10003              9  
1                 NaN              1         10003             21  
2                 NaN              2         10003             15  
3                 NaN              1         10003             15  
4                 NaN              2         10003             34  
...               ...            ...           ...            ...  
186461            NaN              2         10225              1  
186462            NaN              2         10225              1  
186463            NaN              2         10225              1  
186464            NaN              2         10225              1  
186465            NaN              2         10225              1  

[186466 rows x 10 columns]

筛选得到2020年的数据

In [4]:
df1=csv_data[csv_data['year']==2020]
df1

id           majorName  year  minScore  avgScore  maxScore  \
165456   182483                葡萄牙语  2020       391       NaN       NaN   
165459   182486                 朝鲜语  2020       392       NaN       NaN   
165462   182489                  法语  2020       393       NaN       NaN   
165463   182490                  德语  2020       393       NaN       NaN   
165464   182491                西班牙语  2020       393       NaN       NaN   
...         ...                 ...   ...       ...       ...       ...   
186461  5528704  野生动物与自然保护区管理（国家专项）  2020       611       NaN       NaN   
186462  5531969    计算机科学与技术(中外合作办学)  2020       589       NaN       NaN   
186463  5535235          化学(中外合作办学)  2020       577       NaN       NaN   
186464  5538502        生物技术(中外合作办学)  2020       575       NaN       NaN   
186465  5541770        工程管理(中外合作办学)  2020       547       NaN       NaN   

        firstlevelIDs  categoryID_id  collegeID_id  provinceID_id  
165456            NaN              1         10036              4  
165459            NaN              1         10036              4  
165462            NaN              1         10036              4  
165463            NaN              1         10036              4  
165464            NaN              1         10036              4  
...               ...            ...           ...            ...  
186461            NaN              2         10225              1  
186462            NaN              2         10225              1  
186463            NaN              2         10225              1  
186464            NaN              2         10225              1  
186465            NaN              2         10225              1  

[3937 rows x 10 columns]

筛选得到河南省的数据

In [9]:
df1=df1[df1['provinceID_id']==1]
df1

id              majorName  year  minScore  avgScore  maxScore  \
166652   183679                    波斯语  2020       623       NaN       NaN   
166688   183715                   阿拉伯语  2020       624       NaN       NaN   
166689   183716                    朝鲜语  2020       624       NaN       NaN   
166726   183753  中国语言文学类(汉语国际教育、汉语言文学)  2020       625       NaN       NaN   
166727   183754     政治学类(国际政治、政治学与行政学)  2020       625       NaN       NaN   
...         ...                    ...   ...       ...       ...       ...   
186461  5528704     野生动物与自然保护区管理（国家专项）  2020       611       NaN       NaN   
186462  5531969       计算机科学与技术(中外合作办学)  2020       589       NaN       NaN   
186463  5535235             化学(中外合作办学)  2020       577       NaN       NaN   
186464  5538502           生物技术(中外合作办学)  2020       575       NaN       NaN   
186465  5541770           工程管理(中外合作办学)  2020       547       NaN       NaN   

        firstlevelIDs  categoryID_id  collegeID_id  provinceID_id  
166652            NaN              1         10036              1  
166688            NaN              1         10036              1  
166689            NaN              1         10036              1  
166726            NaN              1         10036              1  
166727            NaN              1         10036              1  
...               ...            ...           ...            ...  
186461            NaN              2         10225              1  
186462            NaN              2         10225              1  
186463            NaN              2         10225              1  
186464            NaN              2         10225              1  
186465            NaN              2         10225              1  

[3302 rows x 10 columns]

c1\c2初始化

In [11]:
df1['c1']=None
df1['c2']=None
df1

id              majorName  year  minScore  avgScore  maxScore  \
166652   183679                    波斯语  2020       623       NaN       NaN   
166688   183715                   阿拉伯语  2020       624       NaN       NaN   
166689   183716                    朝鲜语  2020       624       NaN       NaN   
166726   183753  中国语言文学类(汉语国际教育、汉语言文学)  2020       625       NaN       NaN   
166727   183754     政治学类(国际政治、政治学与行政学)  2020       625       NaN       NaN   
...         ...                    ...   ...       ...       ...       ...   
186461  5528704     野生动物与自然保护区管理（国家专项）  2020       611       NaN       NaN   
186462  5531969       计算机科学与技术(中外合作办学)  2020       589       NaN       NaN   
186463  5535235             化学(中外合作办学)  2020       577       NaN       NaN   
186464  5538502           生物技术(中外合作办学)  2020       575       NaN       NaN   
186465  5541770           工程管理(中外合作办学)  2020       547       NaN       NaN   

        firstlevelIDs  categoryID_id  collegeID_id  provinceID_id    c1    c2  
166652            NaN              1         10036              1  None  None  
166688            NaN              1         10036              1  None  None  
166689            NaN              1         10036              1  None  None  
166726            NaN              1         10036              1  None  None  
166727            NaN              1         10036              1  None  None  
...               ...            ...           ...            ...   ...   ...  
186461            NaN              2         10225              1  None  None  
186462            NaN              2         10225              1  None  None  
186463            NaN              2         10225              1  None  None  
186464            NaN              2         10225              1  None  None  
186465            NaN              2         10225              1  None  None  

[3302 rows x 12 columns]

In [13]:
df1=df1.reset_index(drop=True)
df1

id              majorName  year  minScore  avgScore  maxScore  \
0      183679                    波斯语  2020       623       NaN       NaN   
1      183715                   阿拉伯语  2020       624       NaN       NaN   
2      183716                    朝鲜语  2020       624       NaN       NaN   
3      183753  中国语言文学类(汉语国际教育、汉语言文学)  2020       625       NaN       NaN   
4      183754     政治学类(国际政治、政治学与行政学)  2020       625       NaN       NaN   
...       ...                    ...   ...       ...       ...       ...   
3297  5528704     野生动物与自然保护区管理（国家专项）  2020       611       NaN       NaN   
3298  5531969       计算机科学与技术(中外合作办学)  2020       589       NaN       NaN   
3299  5535235             化学(中外合作办学)  2020       577       NaN       NaN   
3300  5538502           生物技术(中外合作办学)  2020       575       NaN       NaN   
3301  5541770           工程管理(中外合作办学)  2020       547       NaN       NaN   

      firstlevelIDs  categoryID_id  collegeID_id  provinceID_id    c1    c2  
0               NaN              1         10036              1  None  None  
1               NaN              1         10036              1  None  None  
2               NaN              1         10036              1  None  None  
3               NaN              1         10036              1  None  None  
4               NaN              1         10036              1  None  None  
...             ...            ...           ...            ...   ...   ...  
3297            NaN              2         10225              1  None  None  
3298            NaN              2         10225              1  None  None  
3299            NaN              2         10225              1  None  None  
3300            NaN              2         10225              1  None  None  
3301            NaN              2         10225              1  None  None  

[3302 rows x 12 columns]

## 开始筛选

 0701 数学

 0702 物理学

 0703 化学

 0704 天文学

 0705 地理学

 0706 大气科学

 0707 海洋科学

 0708 地球物理学

 0709 地质学

 0710 生物学

 0711 系统科学

 0712 科学技术史

 0713 生态学

 0714 统计学

 0801 力学

 0802 机械工程

 0803 光学工程

 0804 仪器科学与技术

 0805 材料科学与工程

 0806 冶金工程

 0807 动力工程及工程热物理

 0808 电气工程

 0809 电子科学与技术

 0810 信息与通信工程

 0811 控制科学与工程

 0812 计算机科学与技术
 计算机类
 √

 0813 建筑学

 0814 土木工程

 0815 水利工程

 0816 测绘科学与技术

 0817 化学工程与技术

 0818 地质资源与地质工程

 0819 矿业工程

 0820 石油与天然气工程

 0821 纺织科学与工程

 0822 轻工技术与工程

 0823 交通运输工程

 0824 船舶与海洋工程

 0825 航空宇航科学与技术

 0826 兵器科学与技术

 0827 核科学与技术

 0828 农业工程

 0829 林业工程

 0830 环境科学与工程

 0831 生物医学工程

 0832 食品科学与工程

 0833 城乡规划学

 0834 风景园林学

 0835 软件工程

 0837 安全科学与工程

1600	1428752	理科试验班类（经济、金融与管理）10003
经济学？

In [839]:
df1[df1['collegeID_id']==10003]

id          majorName  year  minScore  avgScore  maxScore  \
1593  1417804       理科试验班类（新雅书院）  2020       709       NaN       NaN   
1594  1419365       临床医学类（医学实验班）  2020       709       NaN       NaN   
1595  1420927         理科试验班类（数理）  2020       707       NaN       NaN   
1596  1422490               计算机类  2020       706       NaN       NaN   
1597  1424054          临床医学类（协和）  2020       706       NaN       NaN   
1598  1425619      工科试验班类（能源与电气）  2020       705       NaN       NaN   
1599  1427185  工科试验班类（环境、化工与新材料）  2020       705       NaN       NaN   
1600  1428752   理科试验班类（经济、金融与管理）  2020       705       NaN       NaN   
1601  1430320               自动化类  2020       705       NaN       NaN   
1602  1431889   文科试验班类（经济、金融与管理）  2020       672       NaN       NaN   
1603  1433459       文科试验班类（新雅书院）  2020       669       NaN       NaN   

      firstlevelIDs  categoryID_id  collegeID_id  provinceID_id    c1  \
1593            NaN              2         10003              1    理学   
1594            NaN              2         10003              1  None   
1595            NaN              2         10003              1    理学   
1596            NaN              2         10003              1    工学   
1597            NaN              2         10003              1  None   
1598            NaN              2         10003              1    工学   
1599            NaN              2         10003              1    工学   
1600            NaN              2         10003              1  None   
1601            NaN              2         10003              1    工学   
1602            NaN              1         10003              1  None   
1603            NaN              1         10003              1  None   

                       c2  
1593                 None  
1594                 None  
1595             数学类,物理学类  
1596                 计算机类  
1597                 None  
1598            电气类,能源动力类  
1599  材料类,化工与制药类,环境科学与工程类  
1600                 None  
1601                 自动化类  
1602                 None  
1603                 None

In [943]:
df1[df1['majorName']=='公共事业管理']

id majorName  year  minScore  categoryID_id  collegeID_id    c1  \
354    251905    公共事业管理  2020       640              2         10613  None   
357    252874    公共事业管理  2020       613              1         10613  None   
1693  1578854    公共事业管理  2020       579              1         10759  None   
1803  1767559    公共事业管理  2020       619              1         10013  None   
2246  2650015    公共事业管理  2020       598              2         10043  None   
2247  2652229    公共事业管理  2020       610              2         10043  None   
2275  2714627    公共事业管理  2020       632              2         10276  None   
2289  2746120    公共事业管理  2020       613              1         10276  None   
2920  4369052    公共事业管理  2020       596              1         10151  None   

        c2  
354   None  
357   None  
1693  None  
1803  None  
2246  None  
2247  None  
2275  None  
2289  None  
2920  None

In [691]:
df1.loc[i,'majorName']

'工程管理(中外合作办学)'

In [733]:
#df1.loc[0,'c1']==None
index=[]
for i in range(3302):
    if (df1.loc[i,'c1']==None):# and (df1.loc[i,'categoryID_id']==2):
        #if ('医学'not in df1.loc[i,'majorName']) and ('管理'not in df1.loc[i,'majorName']):
        #df1.loc[i,'c1']='工学'
        #df1.loc[i,'c2']='电子信息类'
        index.append(i)
df1.loc[index,:]

id              majorName  year  minScore  avgScore  maxScore  \
0      183679                    波斯语  2020       623       NaN       NaN   
1      183715                   阿拉伯语  2020       624       NaN       NaN   
2      183716                    朝鲜语  2020       624       NaN       NaN   
3      183753  中国语言文学类(汉语国际教育、汉语言文学)  2020       625       NaN       NaN   
4      183754     政治学类(国际政治、政治学与行政学)  2020       625       NaN       NaN   
...       ...                    ...   ...       ...       ...       ...   
3291  5509135             动物科学（国家专项）  2020       613       NaN       NaN   
3295  5522177             物流工程（国家专项）  2020       613       NaN       NaN   
3296  5525440               俄语（国家专项）  2020       611       NaN       NaN   
3297  5528704     野生动物与自然保护区管理（国家专项）  2020       611       NaN       NaN   
3301  5541770           工程管理(中外合作办学)  2020       547       NaN       NaN   

      firstlevelIDs  categoryID_id  collegeID_id  provinceID_id    c1    c2  
0               NaN              1         10036              1  None  None  
1               NaN              1         10036              1  None  None  
2               NaN              1         10036              1  None  None  
3               NaN              1         10036              1  None  None  
4               NaN              1         10036              1  None  None  
...             ...            ...           ...            ...   ...   ...  
3291            NaN              2         10225              1  None  None  
3295            NaN              2         10225              1  None  None  
3296            NaN              2         10225              1  None  None  
3297            NaN              2         10225              1  None  None  
3301            NaN              2         10225              1  None  None  

[1562 rows x 12 columns]

In [ ]:
1740,

In [731]:
df1.loc[index[450:],:]

id          majorName  year  minScore  avgScore  maxScore  \
3045  4737802             俄语（师范）  2020       544       NaN       NaN   
3046  4740815             日语（师范）  2020       544       NaN       NaN   
3047  4743829               商务英语  2020       544       NaN       NaN   
3048  4746844             新闻传播学类  2020       544       NaN       NaN   
3065  4798252               图书馆学  2020       544       NaN       NaN   
3083  4852999            播音与主持艺术  2020       546       NaN       NaN   
3084  4856050                美术学  2020       532       NaN       NaN   
3085  4859102               设计学类  2020       488       NaN       NaN   
3091  4877435         体育教育（公费师范）  2020       483       NaN       NaN   
3093  4883554               经济学类  2020       544       NaN       NaN   
3094  4886615           教育学类（师范）  2020       544       NaN       NaN   
3095  4889677        教育技术学（公费师范）  2020       544       NaN       NaN   
3096  4892740         小学教育（公费师范）  2020       544       NaN       NaN   
3097  4895804           英语（公费师范）  2020       544       NaN       NaN   
3108  4929574               图书馆学  2020       544       NaN       NaN   
3128  4991284       药学类（药学、药物制剂）  2020       608       NaN       NaN   
3129  4994380  经济学类（经济学、国际经济与贸易）  2020       610       NaN       NaN   
3134  5009875        经济学（中外合作办学）  2020       590       NaN       NaN   
3156  5078350                 俄语  2020       654       NaN       NaN   
3169  5119040                 英语  2020       617       NaN       NaN   
3182  5159899                金融学  2020       621       NaN       NaN   
3183  5163049                会计学  2020       609       NaN       NaN   
3192  5191444                 药学  2020       589       NaN       NaN   
3200  5216752           园艺（本硕博班）  2020       610       NaN       NaN   
3204  5229430               金融学类  2020       610       NaN       NaN   
3206  5235775             植物生产类Ⅱ  2020       609       NaN       NaN   
3213  5258014             植物生产类I  2020       607       NaN       NaN   
3218  5273929              动物生产类  2020       606       NaN       NaN   
3219  5277115               动物药学  2020       606       NaN       NaN   
3220  5280302                 园林  2020       606       NaN       NaN   
3223  5289869         自然保护与环境生态类  2020       606       NaN       NaN   
3230  5312227               经济学类  2020       618       NaN       NaN   
3232  5318624         林学类（成栋实验班）  2020       617       NaN       NaN   
3236  5331430         林业工程类（成栋班）  2020       616       NaN       NaN   
3245  5360302               动物科学  2020       614       NaN       NaN   
3248  5369944                林学类  2020       614       NaN       NaN   
3250  5376377                 园林  2020       613       NaN       NaN   
3265  5424752               物流工程  2020       612       NaN       NaN   
3267  5431219                 英语  2020       612       NaN       NaN   
3281  5476600           英语（国家专项）  2020       617       NaN       NaN   
3291  5509135         动物科学（国家专项）  2020       613       NaN       NaN   
3295  5522177         物流工程（国家专项）  2020       613       NaN       NaN   
3296  5525440           俄语（国家专项）  2020       611       NaN       NaN   

      firstlevelIDs  categoryID_id  collegeID_id  provinceID_id    c1    c2  
3045            NaN              2         10200              1  None  None  
3046            NaN              2         10200              1  None  None  
3047            NaN              2         10200              1  None  None  
3048            NaN              2         10200              1  None  None  
3065            NaN              2         10200              1  None  None  
3083            NaN              2         10200              1  None  None  
3084            NaN              2         10200              1  None  None  
3085            NaN              2         10200              1  None  None  
3091            NaN              2         10200              1  None  None  
3093            NaN              2         10200              

In [684]:
df1['c1']

0       None
1       None
2       None
3       None
4       None
        ... 
3297    None
3298      工学
3299      理学
3300      理学
3301    None
Name: c1, Length: 3302, dtype: object

In [687]:
df1[df1['c2']==None]#&(df1['categoryID_id']==1)

Empty DataFrame
Columns: [id, majorName, year, minScore, avgScore, maxScore, firstlevelIDs, categoryID_id, collegeID_id, provinceID_id, c1, c2]
Index: []

In [773]:
a=[2970,3011]
df1.loc[list(a),'c1']='工学'
df1.loc[list(a),'c2']='机械类'
df1.loc[list(a),:]

id                          majorName  year  minScore  avgScore  \
2970  4514677      机械类（含机械工程、工业工程、工程力学、智能制造工程）    2020       650       NaN   
3011  4635955  机械类（含机械工程、工业工程、工程力学、智能制造工程）  国家专项  2020       650       NaN   

      maxScore  firstlevelIDs  categoryID_id  collegeID_id  provinceID_id  c1  \
2970       NaN            NaN              2         10183              1  工学   
3011       NaN            NaN              2         10183              1  工学   

       c2  
2970  机械类  
3011  机械类

In [944]:
#a=df1[df1['majorName'].str.contains('工业工程')].index
a=df1[df1['majorName']=='公共事业管理'].index
df1.loc[list(a),'c1']='管理学'
df1.loc[list(a),'c2']='公共管理类'
df1[df1['majorName']=='公共事业管理']
#df1[df1['majorName'].str.contains('工业工程')]

id majorName  year  minScore  categoryID_id  collegeID_id   c1  \
354    251905    公共事业管理  2020       640              2         10613  管理学   
357    252874    公共事业管理  2020       613              1         10613  管理学   
1693  1578854    公共事业管理  2020       579              1         10759  管理学   
1803  1767559    公共事业管理  2020       619              1         10013  管理学   
2246  2650015    公共事业管理  2020       598              2         10043  管理学   
2247  2652229    公共事业管理  2020       610              2         10043  管理学   
2275  2714627    公共事业管理  2020       632              2         10276  管理学   
2289  2746120    公共事业管理  2020       613              1         10276  管理学   
2920  4369052    公共事业管理  2020       596              1         10151  管理学   

         c2  
354   公共管理类  
357   公共管理类  
1693  公共管理类  
1803  公共管理类  
2246  公共管理类  
2247  公共管理类  
2275  公共管理类  
2289  公共管理类  
2920  公共管理类

dd1954,+=数学类，物理学类，化学类，生物科学类
dd2974，工学	地质类,测绘类
dd2967，工学	材料类,化工与制药类
dd743，化学类
dd741物理学类
dd747生物科学类
dd1224 化学类,生物科学类
dd2959，工学	地质类
dd2992，数学类,统计学类
dd3120，数学类,统计学类
dd1959，统计学类

dd1488 生命科学与技术基地班？工学	生物医学工程类

理科实验班
工科试验班

In [ ]:
df

In [995]:
a=[1600]
df1.loc[list(a),:]

id         majorName  year  minScore  categoryID_id  collegeID_id  \
1600  1428752  理科试验班类（经济、金融与管理）  2020       705              2         10003   

        c1    c2  
1600  None  None

## 匹配找出

In [734]:
df1

id              majorName  year  minScore  avgScore  maxScore  \
0      183679                    波斯语  2020       623       NaN       NaN   
1      183715                   阿拉伯语  2020       624       NaN       NaN   
2      183716                    朝鲜语  2020       624       NaN       NaN   
3      183753  中国语言文学类(汉语国际教育、汉语言文学)  2020       625       NaN       NaN   
4      183754     政治学类(国际政治、政治学与行政学)  2020       625       NaN       NaN   
...       ...                    ...   ...       ...       ...       ...   
3297  5528704     野生动物与自然保护区管理（国家专项）  2020       611       NaN       NaN   
3298  5531969       计算机科学与技术(中外合作办学)  2020       589       NaN       NaN   
3299  5535235             化学(中外合作办学)  2020       577       NaN       NaN   
3300  5538502           生物技术(中外合作办学)  2020       575       NaN       NaN   
3301  5541770           工程管理(中外合作办学)  2020       547       NaN       NaN   

      firstlevelIDs  categoryID_id  collegeID_id  provinceID_id    c1     c2  
0               NaN              1         10036              1  None   None  
1               NaN              1         10036              1  None   None  
2               NaN              1         10036              1  None   None  
3               NaN              1         10036              1  None   None  
4               NaN              1         10036              1  None   None  
...             ...            ...           ...            ...   ...    ...  
3297            NaN              2         10225              1  None   None  
3298            NaN              2         10225              1    工学   计算机类  
3299            NaN              2         10225              1    理学    化学类  
3300            NaN              2         10225              1    理学  生物科学类  
3301            NaN              2         10225              1  None   None  

[3302 rows x 12 columns]

In [757]:
import pandas as pd

csv_file = "7.21/major_left.csv"
#c=['id','majorName','year','minScore','avgScore','maxScore','firstlevelIDs','categoryID_id','collegeID_id','provinceID_id']
df2 = pd.read_csv(csv_file)#,encoding='GBK',names=c)
df2=df2.drop(columns=['Unnamed: 0'])

In [853]:
list(df1[df1['id']==185013].index)

[27]

In [752]:
df1.loc[df1[df1['id']==185013].index[0],'c1']

'工学'

In [758]:
df2

id          majorName  score  c1  c2  c3  collage
0      185013         数据科学与大数据技术    655 NaN NaN NaN    10036
1      200227           地理科学（师范）    599 NaN NaN NaN    10270
2      200239              公共管理类    600 NaN NaN NaN    10270
3      200602              地理科学类    609 NaN NaN NaN    10270
4      200630       电子信息工程（中美合作）    604 NaN NaN NaN    10270
...       ...                ...    ...  ..  ..  ..      ...
1830  5515654  建筑环境与能源应用工程（国家专项）    613 NaN NaN NaN    10225
1831  5518915      木材科学与工程（国家专项）    613 NaN NaN NaN    10225
1832  5531969   计算机科学与技术(中外合作办学)    589 NaN NaN NaN    10225
1833  5535235         化学(中外合作办学)    577 NaN NaN NaN    10225
1834  5538502       生物技术(中外合作办学)    575 NaN NaN NaN    10225

[1835 rows x 7 columns]

In [774]:
No_label=[]
for i in range(1876):
    aid=df2.loc[i,'id']
    index1=df1[df1['id']==aid].index[0]
    if df1.loc[index1,'c1']==None:
        No_label.append(i)
    else:
        df2.loc[i,'c1']=df1.loc[index1,'c1']
        df2.loc[i,'c2']=df1.loc[index1,'c2']
    print(i,'/1876 finished!')

0 /1876 finished!
1 /1876 finished!
2 /1876 finished!
3 /1876 finished!
4 /1876 finished!
5 /1876 finished!
6 /1876 finished!
7 /1876 finished!
8 /1876 finished!
9 /1876 finished!
10 /1876 finished!
11 /1876 finished!
12 /1876 finished!
13 /1876 finished!
14 /1876 finished!
15 /1876 finished!
16 /1876 finished!
17 /1876 finished!
18 /1876 finished!
19 /1876 finished!
20 /1876 finished!
21 /1876 finished!
22 /1876 finished!
23 /1876 finished!
24 /1876 finished!
25 /1876 finished!
26 /1876 finished!
27 /1876 finished!
28 /1876 finished!
29 /1876 finished!
30 /1876 finished!
31 /1876 finished!
32 /1876 finished!
33 /1876 finished!
34 /1876 finished!
35 /1876 finished!
36 /1876 finished!
37 /1876 finished!
38 /1876 finished!
39 /1876 finished!
40 /1876 finished!
41 /1876 finished!
42 /1876 finished!
43 /1876 finished!
44 /1876 finished!
45 /1876 finished!
46 /1876 finished!
47 /1876 finished!
48 /1876 finished!
49 /1876 finished!
50 /1876 finished!
51 /1876 finished!
52 /1876 finished!
53 

429 /1876 finished!
430 /1876 finished!
431 /1876 finished!
432 /1876 finished!
433 /1876 finished!
434 /1876 finished!
435 /1876 finished!
436 /1876 finished!
437 /1876 finished!
438 /1876 finished!
439 /1876 finished!
440 /1876 finished!
441 /1876 finished!
442 /1876 finished!
443 /1876 finished!
444 /1876 finished!
445 /1876 finished!
446 /1876 finished!
447 /1876 finished!
448 /1876 finished!
449 /1876 finished!
450 /1876 finished!
451 /1876 finished!
452 /1876 finished!
453 /1876 finished!
454 /1876 finished!
455 /1876 finished!
456 /1876 finished!
457 /1876 finished!
458 /1876 finished!
459 /1876 finished!
460 /1876 finished!
461 /1876 finished!
462 /1876 finished!
463 /1876 finished!
464 /1876 finished!
465 /1876 finished!
466 /1876 finished!
467 /1876 finished!
468 /1876 finished!
469 /1876 finished!
470 /1876 finished!
471 /1876 finished!
472 /1876 finished!
473 /1876 finished!
474 /1876 finished!
475 /1876 finished!
476 /1876 finished!
477 /1876 finished!
478 /1876 finished!


858 /1876 finished!
859 /1876 finished!
860 /1876 finished!
861 /1876 finished!
862 /1876 finished!
863 /1876 finished!
864 /1876 finished!
865 /1876 finished!
866 /1876 finished!
867 /1876 finished!
868 /1876 finished!
869 /1876 finished!
870 /1876 finished!
871 /1876 finished!
872 /1876 finished!
873 /1876 finished!
874 /1876 finished!
875 /1876 finished!
876 /1876 finished!
877 /1876 finished!
878 /1876 finished!
879 /1876 finished!
880 /1876 finished!
881 /1876 finished!
882 /1876 finished!
883 /1876 finished!
884 /1876 finished!
885 /1876 finished!
886 /1876 finished!
887 /1876 finished!
888 /1876 finished!
889 /1876 finished!
890 /1876 finished!
891 /1876 finished!
892 /1876 finished!
893 /1876 finished!
894 /1876 finished!
895 /1876 finished!
896 /1876 finished!
897 /1876 finished!
898 /1876 finished!
899 /1876 finished!
900 /1876 finished!
901 /1876 finished!
902 /1876 finished!
903 /1876 finished!
904 /1876 finished!
905 /1876 finished!
906 /1876 finished!
907 /1876 finished!


1334 /1876 finished!
1335 /1876 finished!
1336 /1876 finished!
1337 /1876 finished!
1338 /1876 finished!
1339 /1876 finished!
1340 /1876 finished!
1341 /1876 finished!
1342 /1876 finished!
1343 /1876 finished!
1344 /1876 finished!
1345 /1876 finished!
1346 /1876 finished!
1347 /1876 finished!
1348 /1876 finished!
1349 /1876 finished!
1350 /1876 finished!
1351 /1876 finished!
1352 /1876 finished!
1353 /1876 finished!
1354 /1876 finished!
1355 /1876 finished!
1356 /1876 finished!
1357 /1876 finished!
1358 /1876 finished!
1359 /1876 finished!
1360 /1876 finished!
1361 /1876 finished!
1362 /1876 finished!
1363 /1876 finished!
1364 /1876 finished!
1365 /1876 finished!
1366 /1876 finished!
1367 /1876 finished!
1368 /1876 finished!
1369 /1876 finished!
1370 /1876 finished!
1371 /1876 finished!
1372 /1876 finished!
1373 /1876 finished!
1374 /1876 finished!
1375 /1876 finished!
1376 /1876 finished!
1377 /1876 finished!
1378 /1876 finished!
1379 /1876 finished!
1380 /1876 finished!
1381 /1876 fi

1758 /1876 finished!
1759 /1876 finished!
1760 /1876 finished!
1761 /1876 finished!
1762 /1876 finished!
1763 /1876 finished!
1764 /1876 finished!
1765 /1876 finished!
1766 /1876 finished!
1767 /1876 finished!
1768 /1876 finished!
1769 /1876 finished!
1770 /1876 finished!
1771 /1876 finished!
1772 /1876 finished!
1773 /1876 finished!
1774 /1876 finished!
1775 /1876 finished!
1776 /1876 finished!
1777 /1876 finished!
1778 /1876 finished!
1779 /1876 finished!
1780 /1876 finished!
1781 /1876 finished!
1782 /1876 finished!
1783 /1876 finished!
1784 /1876 finished!
1785 /1876 finished!
1786 /1876 finished!
1787 /1876 finished!
1788 /1876 finished!
1789 /1876 finished!
1790 /1876 finished!
1791 /1876 finished!
1792 /1876 finished!
1793 /1876 finished!
1794 /1876 finished!
1795 /1876 finished!
1796 /1876 finished!
1797 /1876 finished!
1798 /1876 finished!
1799 /1876 finished!
1800 /1876 finished!
1801 /1876 finished!
1802 /1876 finished!
1803 /1876 finished!
1804 /1876 finished!
1805 /1876 fi

KeyError: 1835

In [775]:
df2.loc[No_label,:]

id     majorName  score   c1   c2  c3  collage
2      200239         公共管理类    600  NaN  NaN NaN    10270
33     202570  医学技术类（医学四年类）    636  NaN  NaN NaN    10533
34     202780       中国语言文学类    627  NaN  NaN NaN    10533
35     202852       外国语言文学类    627  NaN  NaN NaN    10533
36     202925           all    614  NaN  NaN NaN    10559
...       ...           ...    ...  ...  ...  ..      ...
1777  5235775        植物生产类Ⅱ    609  NaN  NaN NaN    10224
1782  5258014        植物生产类I    607  NaN  NaN NaN    10224
1787  5273929         动物生产类    606  NaN  NaN NaN    10224
1789  5289869    自然保护与环境生态类    606  NaN  NaN NaN    10224
1795  5331430    林业工程类（成栋班）    616  NaN  NaN NaN    10225

[162 rows x 7 columns]

In [763]:
a.to_csv('major_left_left.csv')

In [765]:
pd.read_csv('major_left_left.csv')

Unnamed: 0       id     majorName  score  c1  c2  c3  collage
0            2   200239         公共管理类    600 NaN NaN NaN    10270
1           33   202570  医学技术类（医学四年类）    636 NaN NaN NaN    10533
2           34   202780       中国语言文学类    627 NaN NaN NaN    10533
3           35   202852       外国语言文学类    627 NaN NaN NaN    10533
4           36   202925           all    614 NaN NaN NaN    10559
..         ...      ...           ...    ...  ..  ..  ..      ...
75         930  1460302         公共管理类    666 NaN NaN NaN    10001
76         931  1461890       中国语言文学类    665 NaN NaN NaN    10001
77         934  1469845       外国语言文学类    645 NaN NaN NaN    10004
78         941  1484227       经济管理试验班    647 NaN NaN NaN    10004
79         944  1489039       外国语言文学类    620 NaN NaN NaN    10004

[80 rows x 8 columns]

## 复工

985	1654589	徐特立英才班
1373	3192905	理学工学Ⅱ类
1374	3195352	理学工学Ⅰ类

In [767]:
df3=df2.loc[No_label[80:],:]

In [851]:
df3[50:]

id          majorName  score            c1          c2  c3  collage
1394  3249439            社会科学试验班    639  文学,法学,哲学,历史学         NaN NaN    10255
1406  3299029              新闻传播类    608            文学      新闻传播学类 NaN    10011
1420  3341495              公共管理类    644           管理学       公共管理类 NaN    10251
1422  3354040            外国语言文学类    615            文学     外国语言文学类 NaN    10251
1423  3356552              公共管理类    612           管理学       公共管理类 NaN    10251
1448  3422215            焊接技术与工程    614            工学         机械类 NaN    10291
1453  3445102              公共管理类    594           管理学       公共管理类 NaN    10291
1454  3462959            数据管理与应用    616           管理学    管理科学与工程类 NaN    10327
1460  3527134            经济管理试验班    682           经济学        经济学类 NaN    10247
1465  3540044            人文科学试验班    635  文学,法学,哲学,历史学         NaN NaN    10247
1466  3542629            社会科学试验班    638  文学,法学,哲学,历史学         NaN NaN    10247
1511  3686344              公共管理类    637           管理学       公共管理类 NaN    10635
1520  3726064         自然保护与环境生态类    632            农学  自然保护与环境生态类 NaN    10635
1524  3742015              植物生产类    631            农学       植物生产类 NaN    10635
1526  3763339            中国语言文学类    620            文学     中国语言文学类 NaN    10635
1528  3779374              公共管理类    617           管理学       公共管理类 NaN    10635
1549  3876340              公共管理类    654           管理学       公共管理类 NaN   106111
1550  3879052            人文科学试验班    630  文学,法学,哲学,历史学         NaN NaN   106111
1551  3892627              公共管理类    628           管理学       公共管理类 NaN   106111
1552  3895345            外国语言文学类    627            文学     外国语言文学类 NaN   106111
1680  4476574              朝鲜语      652            文学     外国语言文学类 NaN    10183
1724  4859102               设计学类    488           艺术学        设计学类 NaN    10200
1725  4862155             广播电视编导    590           艺术学     戏剧与影视学类 NaN    10200
1726  4874377               设计学类    488           艺术学        设计学类 NaN    10200
1748  5053394      经济管理试验班（国家专项）    663           经济学        经济学类 NaN    10213
1756  5084599  经济管理试验班（含AI联合培养班）    625           经济学        经济学类 NaN    10213
1757  5090852               设计学类    623           艺术学        设计学类 NaN    10213
1777  5235775             植物生产类Ⅱ    609            农学       植物生产类 NaN    10224
1782  5258014             植物生产类I    607            农学       植物生产类 NaN    10224
1787  5273929              动物生产类    606            农学       动物生产类 NaN    10224
1789  5289869         自然保护与环境生态类    606            农学  自然保护与环境生态类 NaN    10224
1795  5331430         林业工程类（成栋班）    616            工学       林业工程类 NaN    10225

In [846]:
df3[df3['majorName'].str.contains('人文')]

id majorName  score  c1   c2  c3  collage
982   1649477   人文科学试验班    655  文学  NaN NaN    10002
1375  3197800     人文社科类    646  文学  NaN NaN    10280
1378  3207602     人文社科类    622  文学  NaN NaN    10280
1465  3540044   人文科学试验班    635  文学  NaN NaN    10247
1550  3879052   人文科学试验班    630  文学  NaN NaN   106111

In [847]:
a=df3[df3['majorName'].str.contains('人文')].index
#a=df3[df3['majorName']=='社会科学试验班'].index
df3.loc[list(a),'c1']='文学,法学,哲学,历史学'
#df3.loc[list(a),'c2']='经济与贸易类'
#df3[df3['majorName']=='社会科学试验班']
df3[df3['majorName'].str.contains('人文')]

id majorName  score            c1   c2  c3  collage
982   1649477   人文科学试验班    655  文学,法学,哲学,历史学  NaN NaN    10002
1375  3197800     人文社科类    646  文学,法学,哲学,历史学  NaN NaN    10280
1378  3207602     人文社科类    622  文学,法学,哲学,历史学  NaN NaN    10280
1465  3540044   人文科学试验班    635  文学,法学,哲学,历史学  NaN NaN    10247
1550  3879052   人文科学试验班    630  文学,法学,哲学,历史学  NaN NaN   106111

In [909]:
a==df3.loc[982,'c2']

False

In [850]:
a=[1373,1374]
df3.loc[list(a),'c1']='工学,理学'
#df3.loc[list(a),'c2']='外国语言文学类'
df3.loc[list(a),:]

id majorName  score     c1   c2  c3  collage
1373  3192905    理学工学Ⅱ类    643  工学,理学  NaN NaN    10280
1374  3195352    理学工学Ⅰ类    643  工学,理学  NaN NaN    10280

## 汇总

In [936]:
import pandas as pd

csv_file = "7.21/major(1).csv"
#c=['id','majorName','year','minScore','avgScore','maxScore','firstlevelIDs','categoryID_id','collegeID_id','provinceID_id']
df4 = pd.read_csv(csv_file)#,encoding='GBK',names=c)
df4=df4.drop(columns=['Unnamed: 0'])

In [937]:
df4=[df4,df3]
df4=pd.concat(df4,ignore_index=True)
df4=df4.drop(columns=['c3'])
df4

id           majorName  score  c1          c2  collage
0      183754  政治学类(国际政治、政治学与行政学)    625  法学        政治学类    10036
1      183942             法学类(法学)    630  法学         法学类    10036
2      184909             法学类(法学)    653  法学         法学类    10036
3      200230                  法学    602  法学         法学类    10270
4      200377                社会工作    599  法学        社会学类    10270
...       ...                 ...    ...  ..         ...      ...
1791  5235775              植物生产类Ⅱ    609  农学       植物生产类    10224
1792  5258014              植物生产类I    607  农学       植物生产类    10224
1793  5273929               动物生产类    606  农学       动物生产类    10224
1794  5289869          自然保护与环境生态类    606  农学  自然保护与环境生态类    10224
1795  5331430          林业工程类（成栋班）    616  工学       林业工程类    10225

[1796 rows x 6 columns]

In [938]:
df4=df4.sort_values(by='id',ignore_index=True)
df4

id              majorName  score   c1          c2  collage
0      183679                    波斯语    623   文学     外国语言文学类    10036
1      183715                   阿拉伯语    624   文学     外国语言文学类    10036
2      183716                    朝鲜语    624   文学     外国语言文学类    10036
3      183753  中国语言文学类(汉语国际教育、汉语言文学)    625   文学     中国语言文学类    10036
4      183754     政治学类(国际政治、政治学与行政学)    625   法学        政治学类    10036
...       ...                    ...    ...  ...         ...      ...
1791  5509135             动物科学（国家专项）    613   农学       动物生产类    10225
1792  5522177             物流工程（国家专项）    613  管理学    物流管理与工程类    10225
1793  5525440               俄语（国家专项）    611   文学     外国语言文学类    10225
1794  5528704     野生动物与自然保护区管理（国家专项）    611   农学  自然保护与环境生态类    10225
1795  5541770           工程管理(中外合作办学)    547  管理学    管理科学与工程类    10225

[1796 rows x 6 columns]

In [881]:
df1=df1.drop(columns=['provinceID_id'])
df1

id              majorName  year  minScore  categoryID_id  \
0      183679                    波斯语  2020       623              1   
1      183715                   阿拉伯语  2020       624              1   
2      183716                    朝鲜语  2020       624              1   
3      183753  中国语言文学类(汉语国际教育、汉语言文学)  2020       625              1   
4      183754     政治学类(国际政治、政治学与行政学)  2020       625              1   
...       ...                    ...   ...       ...            ...   
3297  5528704     野生动物与自然保护区管理（国家专项）  2020       611              2   
3298  5531969       计算机科学与技术(中外合作办学)  2020       589              2   
3299  5535235             化学(中外合作办学)  2020       577              2   
3300  5538502           生物技术(中外合作办学)  2020       575              2   
3301  5541770           工程管理(中外合作办学)  2020       547              2   

      collegeID_id    c1     c2  
0            10036  None   None  
1            10036  None   None  
2            10036  None   None  
3            10036  None   None  
4            10036  None   None  
...            ...   ...    ...  
3297         10225  None   None  
3298         10225    工学   计算机类  
3299         10225    理学    化学类  
3300         10225    理学  生物科学类  
3301         10225  None   None  

[3302 rows x 8 columns]

In [895]:
new=df1.loc[0,:]
new[['c1','c2']]=['a','b']
print(new,df1.loc[0,:])

id               183679
majorName           波斯语
year               2020
minScore            623
categoryID_id         1
collegeID_id      10036
c1                    a
c2                    b
Name: 0, dtype: object id               183679
majorName           波斯语
year               2020
minScore            623
categoryID_id         1
collegeID_id      10036
c1                 None
c2                 None
Name: 0, dtype: object


E:\anaconda\lib\site-packages\pandas\core\indexing.py:1965: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self.obj._check_is_chained_assignment_possible()
E:\anaconda\lib\site-packages\pandas\core\indexing.py:1732: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_block(indexer, value, name)


In [916]:
type(df4.loc[160,'c2'])==str

True

In [911]:
df4[160:200]

id                   majorName  score   c1          c2  collage  \
160  232609                          日语    624   文学     外国语言文学类    10532   
161  232864                     人文科学试验班    624   文学         NaN    10532   
162  233120                        经济学类    626  经济学        经济学类    10532   
163  233377                         金融学    628  经济学        金融学类    10532   
164  233635                         会计学    629  管理学       工商管理类    10532   
165  233894                       汉语言文学    625   文学     中国语言文学类    10532   
166  234154                     国际经济与贸易    628  经济学      经济与贸易类    10532   
167  234415                        工商管理    625  管理学       工商管理类    10532   
168  234677                          法学    626   法学         法学类    10532   
169  234940                体育教育(中外合作办学)    428  教育学        体育学类    10542   
170  235204                          法学    630   法学         法学类    10542   
171  235469                        知识产权    630   法学         法学类    10542   
172  239005                      土地资源管理    624  管理学       公共管理类    10542   
173  239284                        工商管理    623  管理学       工商管理类    10542   
174  239845                体育教育(中外合作办学)    451  教育学        体育学类    10542   
175  240410                          法学    614   法学         法学类    10542   
176  240694                      思想政治教育    612   法学    马克思主义理论类    10542   
177  240979                         历史学    611  历史学        历史学类    10542   
178  241265                       汉语言文学    613   文学     中国语言文学类    10542   
179  241552                         新闻学    612   文学      新闻传播学类    10542   
180  241840                       编辑出版学    611   文学      新闻传播学类    10542   
181  242710                          园林    562   农学         林学类    10389   
182  243002                          园艺    562   农学       植物生产类    10389   
183  243295                        风景园林    598   农学         林学类    10389   
184  244180   农业经济管理类（含农林经济管理、农村区域发展专业）    554  管理学     农业经济管理类    10389   
185  244477                          茶学    528   农学       植物生产类    10389   
186  244775                          蜂学    535   农学       动物生产类    10389   
187  245074  林学类（含林学、森林保护、水土保持与荒漠化防治专业）    537   农学         林学类    10389   
188  245074  林学类（含林学、森林保护、水土保持与荒漠化防治专业）    537   农学  自然保护与环境生态类    10389   
189  245675                        动物科学    534   农学       动物生产类    10389   
190  245977                        管理科学    535  管理学    管理科学与工程类    10389   
191  246584                        植物保护    536   农学       植物生产类    10389   
192  246889                       商务经济学    544  经济学        经济学类    10389   
193  247502                       水产养殖学    534   农学         水产类    10389   
194  248119                     植物科学与技术    534   农学       植物生产类    10389   
195  248429                   设施农业科学与工程    535   农学       植物生产类    10389   
196  249052                          茶学    538   农学       植物生产类    10389   
197  249365                       商务经济学    548  经济学        经济学类    10389   
198  249679                   园艺（中加合作班）    529   农学       植物生产类    10389   
199  249994                 风景园林（中加合作班）    540   农学         林学类    10389   

    category  
160     None  
161     None  
162     None  
163     None  
164     None  
165     None  
166     None  
167     None  
168     None  
169     None  
170     None  
171     None  
172     None  
173     None  
174     None  
175     None  
176     None  
177     None  
178     None  
179     None  
180     None  
181     None  
182     None  
183     None  
184     None  
185     None  
186     None  
187     None  
188     None  
189     None  
190     None  
191     None  
192     None  
193     None  
194     None  
195     None  
196     None  
197     None  
198     None  
199     None

In [996]:
df5=df1.loc[[0,1],:]
df5=pd.DataFrame(df5)
df5.loc[[0,1],['c1','c2']]=df4.loc[[0,1],['c1','c2']]
df5

id majorName  year  minScore  categoryID_id  collegeID_id  c1       c2
0  183679       波斯语  2020       623              1         10036  文学  外国语言文学类
1  183715      阿拉伯语  2020       624              1         10036  文学  外国语言文学类

In [997]:
No_label=[]
Do_label=[]
#df5=df5.append(df4.loc[2,:])

for i in range(2,3302):
    aid=df1.loc[i,'id']
    indexes=list(df4[df4['id']==aid].index)
    if len(indexes)==0:#未在df4标记中
        if df1.loc[i,'c1']==None:#也不在df1标记中
            No_label.append(i)
        else:#在df1标记中
            df5=df5.append(df1.loc[i,:])
    else:#在df4标记中
        if df1.loc[i,'c1']==None:#不在df1标记中
            for ind in indexes:#可能有多个标记
                new=df1.loc[i,:]
                new['c1']=df4.loc[ind,'c1']
                if type(df4.loc[ind,'c2'])==str:
                    new['c2']=df4.loc[ind,'c2']
                df5=df5.append(new)
        else:#也在df1标记中,以df1为准
            df5=df5.append(df1.loc[i,:])
            Do_label.append(i)
    
    print(i,'/ 3301  finished!')

E:\anaconda\lib\site-packages\pandas\core\series.py:1056: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  cacher_needs_updating = self._check_is_chained_assignment_possible()


2 / 3301  finished!
3 / 3301  finished!
4 / 3301  finished!
5 / 3301  finished!
6 / 3301  finished!
7 / 3301  finished!
8 / 3301  finished!
9 / 3301  finished!
10 / 3301  finished!
11 / 3301  finished!
12 / 3301  finished!
13 / 3301  finished!
14 / 3301  finished!
15 / 3301  finished!
16 / 3301  finished!
17 / 3301  finished!
18 / 3301  finished!
19 / 3301  finished!
20 / 3301  finished!
21 / 3301  finished!
22 / 3301  finished!
23 / 3301  finished!
24 / 3301  finished!
25 / 3301  finished!
26 / 3301  finished!
27 / 3301  finished!
28 / 3301  finished!
29 / 3301  finished!
30 / 3301  finished!
31 / 3301  finished!
32 / 3301  finished!
33 / 3301  finished!
34 / 3301  finished!
35 / 3301  finished!
36 / 3301  finished!
37 / 3301  finished!
38 / 3301  finished!
39 / 3301  finished!
40 / 3301  finished!
41 / 3301  finished!
42 / 3301  finished!
43 / 3301  finished!
44 / 3301  finished!
45 / 3301  finished!
46 / 3301  finished!
47 / 3301  finished!
48 / 3301  finished!
49 / 3301  finished!


383 / 3301  finished!
384 / 3301  finished!
385 / 3301  finished!
386 / 3301  finished!
387 / 3301  finished!
388 / 3301  finished!
389 / 3301  finished!
390 / 3301  finished!
391 / 3301  finished!
392 / 3301  finished!
393 / 3301  finished!
394 / 3301  finished!
395 / 3301  finished!
396 / 3301  finished!
397 / 3301  finished!
398 / 3301  finished!
399 / 3301  finished!
400 / 3301  finished!
401 / 3301  finished!
402 / 3301  finished!
403 / 3301  finished!
404 / 3301  finished!
405 / 3301  finished!
406 / 3301  finished!
407 / 3301  finished!
408 / 3301  finished!
409 / 3301  finished!
410 / 3301  finished!
411 / 3301  finished!
412 / 3301  finished!
413 / 3301  finished!
414 / 3301  finished!
415 / 3301  finished!
416 / 3301  finished!
417 / 3301  finished!
418 / 3301  finished!
419 / 3301  finished!
420 / 3301  finished!
421 / 3301  finished!
422 / 3301  finished!
423 / 3301  finished!
424 / 3301  finished!
425 / 3301  finished!
426 / 3301  finished!
427 / 3301  finished!
428 / 3301

813 / 3301  finished!
814 / 3301  finished!
815 / 3301  finished!
816 / 3301  finished!
817 / 3301  finished!
818 / 3301  finished!
819 / 3301  finished!
820 / 3301  finished!
821 / 3301  finished!
822 / 3301  finished!
823 / 3301  finished!
824 / 3301  finished!
825 / 3301  finished!
826 / 3301  finished!
827 / 3301  finished!
828 / 3301  finished!
829 / 3301  finished!
830 / 3301  finished!
831 / 3301  finished!
832 / 3301  finished!
833 / 3301  finished!
834 / 3301  finished!
835 / 3301  finished!
836 / 3301  finished!
837 / 3301  finished!
838 / 3301  finished!
839 / 3301  finished!
840 / 3301  finished!
841 / 3301  finished!
842 / 3301  finished!
843 / 3301  finished!
844 / 3301  finished!
845 / 3301  finished!
846 / 3301  finished!
847 / 3301  finished!
848 / 3301  finished!
849 / 3301  finished!
850 / 3301  finished!
851 / 3301  finished!
852 / 3301  finished!
853 / 3301  finished!
854 / 3301  finished!
855 / 3301  finished!
856 / 3301  finished!
857 / 3301  finished!
858 / 3301

1225 / 3301  finished!
1226 / 3301  finished!
1227 / 3301  finished!
1228 / 3301  finished!
1229 / 3301  finished!
1230 / 3301  finished!
1231 / 3301  finished!
1232 / 3301  finished!
1233 / 3301  finished!
1234 / 3301  finished!
1235 / 3301  finished!
1236 / 3301  finished!
1237 / 3301  finished!
1238 / 3301  finished!
1239 / 3301  finished!
1240 / 3301  finished!
1241 / 3301  finished!
1242 / 3301  finished!
1243 / 3301  finished!
1244 / 3301  finished!
1245 / 3301  finished!
1246 / 3301  finished!
1247 / 3301  finished!
1248 / 3301  finished!
1249 / 3301  finished!
1250 / 3301  finished!
1251 / 3301  finished!
1252 / 3301  finished!
1253 / 3301  finished!
1254 / 3301  finished!
1255 / 3301  finished!
1256 / 3301  finished!
1257 / 3301  finished!
1258 / 3301  finished!
1259 / 3301  finished!
1260 / 3301  finished!
1261 / 3301  finished!
1262 / 3301  finished!
1263 / 3301  finished!
1264 / 3301  finished!
1265 / 3301  finished!
1266 / 3301  finished!
1267 / 3301  finished!
1268 / 3301

1633 / 3301  finished!
1634 / 3301  finished!
1635 / 3301  finished!
1636 / 3301  finished!
1637 / 3301  finished!
1638 / 3301  finished!
1639 / 3301  finished!
1640 / 3301  finished!
1641 / 3301  finished!
1642 / 3301  finished!
1643 / 3301  finished!
1644 / 3301  finished!
1645 / 3301  finished!
1646 / 3301  finished!
1647 / 3301  finished!
1648 / 3301  finished!
1649 / 3301  finished!
1650 / 3301  finished!
1651 / 3301  finished!
1652 / 3301  finished!
1653 / 3301  finished!
1654 / 3301  finished!
1655 / 3301  finished!
1656 / 3301  finished!
1657 / 3301  finished!
1658 / 3301  finished!
1659 / 3301  finished!
1660 / 3301  finished!
1661 / 3301  finished!
1662 / 3301  finished!
1663 / 3301  finished!
1664 / 3301  finished!
1665 / 3301  finished!
1666 / 3301  finished!
1667 / 3301  finished!
1668 / 3301  finished!
1669 / 3301  finished!
1670 / 3301  finished!
1671 / 3301  finished!
1672 / 3301  finished!
1673 / 3301  finished!
1674 / 3301  finished!
1675 / 3301  finished!
1676 / 3301

1991 / 3301  finished!
1992 / 3301  finished!
1993 / 3301  finished!
1994 / 3301  finished!
1995 / 3301  finished!
1996 / 3301  finished!
1997 / 3301  finished!
1998 / 3301  finished!
1999 / 3301  finished!
2000 / 3301  finished!
2001 / 3301  finished!
2002 / 3301  finished!
2003 / 3301  finished!
2004 / 3301  finished!
2005 / 3301  finished!
2006 / 3301  finished!
2007 / 3301  finished!
2008 / 3301  finished!
2009 / 3301  finished!
2010 / 3301  finished!
2011 / 3301  finished!
2012 / 3301  finished!
2013 / 3301  finished!
2014 / 3301  finished!
2015 / 3301  finished!
2016 / 3301  finished!
2017 / 3301  finished!
2018 / 3301  finished!
2019 / 3301  finished!
2020 / 3301  finished!
2021 / 3301  finished!
2022 / 3301  finished!
2023 / 3301  finished!
2024 / 3301  finished!
2025 / 3301  finished!
2026 / 3301  finished!
2027 / 3301  finished!
2028 / 3301  finished!
2029 / 3301  finished!
2030 / 3301  finished!
2031 / 3301  finished!
2032 / 3301  finished!
2033 / 3301  finished!
2034 / 3301

2369 / 3301  finished!
2370 / 3301  finished!
2371 / 3301  finished!
2372 / 3301  finished!
2373 / 3301  finished!
2374 / 3301  finished!
2375 / 3301  finished!
2376 / 3301  finished!
2377 / 3301  finished!
2378 / 3301  finished!
2379 / 3301  finished!
2380 / 3301  finished!
2381 / 3301  finished!
2382 / 3301  finished!
2383 / 3301  finished!
2384 / 3301  finished!
2385 / 3301  finished!
2386 / 3301  finished!
2387 / 3301  finished!
2388 / 3301  finished!
2389 / 3301  finished!
2390 / 3301  finished!
2391 / 3301  finished!
2392 / 3301  finished!
2393 / 3301  finished!
2394 / 3301  finished!
2395 / 3301  finished!
2396 / 3301  finished!
2397 / 3301  finished!
2398 / 3301  finished!
2399 / 3301  finished!
2400 / 3301  finished!
2401 / 3301  finished!
2402 / 3301  finished!
2403 / 3301  finished!
2404 / 3301  finished!
2405 / 3301  finished!
2406 / 3301  finished!
2407 / 3301  finished!
2408 / 3301  finished!
2409 / 3301  finished!
2410 / 3301  finished!
2411 / 3301  finished!
2412 / 3301

2745 / 3301  finished!
2746 / 3301  finished!
2747 / 3301  finished!
2748 / 3301  finished!
2749 / 3301  finished!
2750 / 3301  finished!
2751 / 3301  finished!
2752 / 3301  finished!
2753 / 3301  finished!
2754 / 3301  finished!
2755 / 3301  finished!
2756 / 3301  finished!
2757 / 3301  finished!
2758 / 3301  finished!
2759 / 3301  finished!
2760 / 3301  finished!
2761 / 3301  finished!
2762 / 3301  finished!
2763 / 3301  finished!
2764 / 3301  finished!
2765 / 3301  finished!
2766 / 3301  finished!
2767 / 3301  finished!
2768 / 3301  finished!
2769 / 3301  finished!
2770 / 3301  finished!
2771 / 3301  finished!
2772 / 3301  finished!
2773 / 3301  finished!
2774 / 3301  finished!
2775 / 3301  finished!
2776 / 3301  finished!
2777 / 3301  finished!
2778 / 3301  finished!
2779 / 3301  finished!
2780 / 3301  finished!
2781 / 3301  finished!
2782 / 3301  finished!
2783 / 3301  finished!
2784 / 3301  finished!
2785 / 3301  finished!
2786 / 3301  finished!
2787 / 3301  finished!
2788 / 3301

3117 / 3301  finished!
3118 / 3301  finished!
3119 / 3301  finished!
3120 / 3301  finished!
3121 / 3301  finished!
3122 / 3301  finished!
3123 / 3301  finished!
3124 / 3301  finished!
3125 / 3301  finished!
3126 / 3301  finished!
3127 / 3301  finished!
3128 / 3301  finished!
3129 / 3301  finished!
3130 / 3301  finished!
3131 / 3301  finished!
3132 / 3301  finished!
3133 / 3301  finished!
3134 / 3301  finished!
3135 / 3301  finished!
3136 / 3301  finished!
3137 / 3301  finished!
3138 / 3301  finished!
3139 / 3301  finished!
3140 / 3301  finished!
3141 / 3301  finished!
3142 / 3301  finished!
3143 / 3301  finished!
3144 / 3301  finished!
3145 / 3301  finished!
3146 / 3301  finished!
3147 / 3301  finished!
3148 / 3301  finished!
3149 / 3301  finished!
3150 / 3301  finished!
3151 / 3301  finished!
3152 / 3301  finished!
3153 / 3301  finished!
3154 / 3301  finished!
3155 / 3301  finished!
3156 / 3301  finished!
3157 / 3301  finished!
3158 / 3301  finished!
3159 / 3301  finished!
3160 / 3301

In [998]:
df5

id              majorName  year  minScore  categoryID_id  \
0      183679                    波斯语  2020       623              1   
1      183715                   阿拉伯语  2020       624              1   
2      183716                    朝鲜语  2020       624              1   
3      183753  中国语言文学类(汉语国际教育、汉语言文学)  2020       625              1   
4      183754     政治学类(国际政治、政治学与行政学)  2020       625              1   
...       ...                    ...   ...       ...            ...   
3297  5528704     野生动物与自然保护区管理（国家专项）  2020       611              2   
3298  5531969       计算机科学与技术(中外合作办学)  2020       589              2   
3299  5535235             化学(中外合作办学)  2020       577              2   
3300  5538502           生物技术(中外合作办学)  2020       575              2   
3301  5541770           工程管理(中外合作办学)  2020       547              2   

      collegeID_id   c1          c2  
0            10036   文学     外国语言文学类  
1            10036   文学     外国语言文学类  
2            10036   文学     外国语言文学类  
3            10036   文学     中国语言文学类  
4            10036   法学        政治学类  
...            ...  ...         ...  
3297         10225   农学  自然保护与环境生态类  
3298         10225   工学        计算机类  
3299         10225   理学         化学类  
3300         10225   理学       生物科学类  
3301         10225  管理学    管理科学与工程类  

[3476 rows x 8 columns]

In [1000]:
df1.loc[No_label,:]

id majorName  year  minScore  categoryID_id  collegeID_id    c1    c2
106  202925       all  2020       614              1         10559  None  None
107  202999       all  2020       636              2         10559  None  None
239  221545       all  2020       618              2         10593  None  None
240  221752       all  2020       600              1         10593  None  None
313  239564      高校专项  2020       571              2         10542  None  None
322  242129      高校专项  2020       568              1         10542  None  None

In [1002]:
df1[df1['collegeID_id']==10593]

id majorName  year  minScore  categoryID_id  collegeID_id    c1    c2
239  221545       all  2020       618              2         10593  None  None
240  221752       all  2020       600              1         10593  None  None

In [957]:
idlist=list(df1.loc[Do_label,:]['id'].values)

In [958]:
idlist=list(df1.loc[Do_label,:]['id'].values)
dl_index=[]
for sid in idlist:
    dl_index.extend(list(df4[df4['id']==sid].index))
dl_index

[97,
 100,
 125,
 137,
 150,
 183,
 184,
 195,
 199,
 200,
 202,
 203,
 207,
 213,
 216,
 242,
 321,
 322,
 326,
 372,
 400,
 406,
 413,
 428,
 449,
 505,
 519,
 536,
 547,
 564,
 685,
 730,
 762,
 764,
 767,
 821,
 831,
 999,
 1049,
 1097,
 1131,
 1218,
 1369,
 1425,
 1459,
 1461,
 1462,
 1491,
 1493,
 1494,
 1495,
 1499,
 1561,
 1562,
 1570,
 1571,
 1585,
 1586,
 1598,
 1599,
 1600,
 1601,
 1606,
 1607,
 1742,
 1748,
 1749,
 1763,
 1771,
 1789]

In [987]:
df4.loc[dl_index,:][35:]

id                                          majorName  score   c1  \
821   1511614                                               工业工程    588  管理学   
831   1537654                                            农业资源与环境    587   农学   
999   1979665                                       计算机科学与技术（英语）    654   文学   
1049  2151524                                               风景园林    641   农学   
1097  2320994                                               工业工程    594  管理学   
1131  2474002                                               工业工程    594  管理学   
1218  2786899                                               工业工程    609  管理学   
1369  3432377                                               风景园林    614   农学   
1425  3718102                                               风景园林    633   农学   
1459  3846574                                               风景园林    657   农学   
1461  3870919                                     生物医学工程类（生物与药学）    654   医学   
1462  3870919                                     生物医学工程类（生物与药学）    654   医学   
1491  3985600                                         车辆工程（英语强化）    655   文学   
1493  4026985                                  机械设计制造及其自动化（日语强化）    654   文学   
1494  4032520                                       金属材料工程（日语强化）    652   文学   
1495  4054700                               车辆工程（英语强化）（贫困地区专项计划）    653   文学   
1499  4124425                                               风景园林    609   农学   
1561  4482425                          地质类（含地质学、资源勘查工程、土地资源管理）      646  管理学   
1562  4485352  电子信息类（含电子信息工程、电子科学与技术、微电子科学与工程、电子信息科学与技术、生物医学工...    652   医学   
1570  4514677                      机械类（含机械工程、工业工程、工程力学、智能制造工程）      650  管理学   
1571  4520554          交通运输类（含交通运输、交通工程、汽车服务工程、道路桥梁与渡河工程、物流工程）      646  管理学   
1585  4567714                           生物工程类（含生物工程、生物制药、药物制剂）      646   医学   
1586  4579544                   数学类（含数学与应用数学、信息与计算科学、金融数学、统计学）      653  经济学   
1598  4635955                  机械类（含机械工程、工业工程、工程力学、智能制造工程）  国家专项    650  管理学   
1599  4641914      交通运输类（含交通运输、交通工程、汽车服务工程、道路桥梁与渡河工程、物流工程）  国家专项    645  管理学   
1600  4650860                   植物生产类（含农学、园艺、植物保护、农业资源与环境）  国家专项    638   农学   
1601  4650860                   植物生产类（含农学、园艺、植物保护、农业资源与环境）  国家专项    638   农学   
1606  4659815                   植物生产类（含农学、园艺、植物保护、农业资源与环境）  农学单列    620   农学   
1607  4659815                   植物生产类（含农学、园艺、植物保护、农业资源与环境）  农学单列    620   农学   
1742  5200927                                      农业资源与环境（本硕博班）    615   农学   
1748  5248477                                               风景园林    608   农学   
1749  5254834                                              工业工程类    607  管理学   
1763  5325025                                               风景园林    617   农学   
1771  5382814                                               工业工程    613  管理学   
1789  5499364                                         工业工程（国家专项）    614  管理学   

              c2  collage  
821        工业工程类    10759  
831   自然保护与环境生态类    10759  
999      外国语言文学类    10030  
1049         林学类    10022  
1097       工业工程类    10755  
1131       工业工程类    10755  
1218       工业工程类    10216  
1369         林学类    10291  
1425         林学类    10635  
1459         林学类   106111  
1461         药学类   106111  
1462       基础医学类   106111  
1491     外国语言文学类    10141  
1493     外国语言文学类    10141  
1494     外国语言文学类    10141  
1495     外国语言文学类    10141  
1499         林学类    11078  
1561       公共管理类    10183  
1562       基础医学类    10183  
1570       工业工程类    10183  
1571    物流管理与工程类    10183  
1585         药学类    10183  
1586        金融学类    10183  
1598       工业工程类    10183  
1599    物流管理与工程类    10183  
1600       植物生产类    10183  
1601  自然保护与环境生态类    10183  
1606  自然保护与环境生态类    10183  
1607       植物生产类    10183  
1742  自然保护与环境生态类    10224  
1748         林学类    10224  
1749       工业工程类    10224  
1763         林学类    10225  
1771       工业工程类    10225  
1789       工业工程类    10225

In [993]:
a=[3195]
df1.loc[a,'c1']='农学'
df1.loc[a,'c2']='自然保护与环境生态类'#,植物生产类'
df1.loc[a,:]

id      majorName  year  minScore  categoryID_id  collegeID_id  c1  \
3195  5200927  农业资源与环境（本硕博班）  2020       615              2         10224  农学   

              c2  
3195  自然保护与环境生态类

In [986]:
df1.loc[Do_label,:][35:]

id                                          majorName  year  \
1652  1511614                                               工业工程  2020   
1668  1537654                                            农业资源与环境  2020   
1919  1979665                                       计算机科学与技术（英语）  2020   
2008  2151524                                               风景园林  2020   
2092  2320994                                               工业工程  2020   
2165  2474002                                               工业工程  2020   
2307  2786899                                               工业工程  2020   
2575  3432377                                               风景园林  2020   
2685  3718102                                               风景园林  2020   
2733  3846574                                               风景园林  2020   
2742  3870919                                     生物医学工程类（生物与药学）  2020   
2784  3985600                                         车辆工程（英语强化）  2020   
2799  4026985                                  机械设计制造及其自动化（日语强化）  2020   
2801  4032520                                       金属材料工程（日语强化）  2020   
2809  4054700                               车辆工程（英语强化）（贫困地区专项计划）  2020   
2834  4124425                                               风景园林  2020   
2959  4482425                          地质类（含地质学、资源勘查工程、土地资源管理）    2020   
2960  4485352  电子信息类（含电子信息工程、电子科学与技术、微电子科学与工程、电子信息科学与技术、生物医学工...  2020   
2970  4514677                      机械类（含机械工程、工业工程、工程力学、智能制造工程）    2020   
2972  4520554          交通运输类（含交通运输、交通工程、汽车服务工程、道路桥梁与渡河工程、物流工程）    2020   
2988  4567714                           生物工程类（含生物工程、生物制药、药物制剂）    2020   
2992  4579544                   数学类（含数学与应用数学、信息与计算科学、金融数学、统计学）    2020   
3011  4635955                  机械类（含机械工程、工业工程、工程力学、智能制造工程）  国家专项  2020   
3013  4641914      交通运输类（含交通运输、交通工程、汽车服务工程、道路桥梁与渡河工程、物流工程）  国家专项  2020   
3016  4650860                   植物生产类（含农学、园艺、植物保护、农业资源与环境）  国家专项  2020   
3019  4659815                   植物生产类（含农学、园艺、植物保护、农业资源与环境）  农学单列  2020   
3195  5200927                                      农业资源与环境（本硕博班）  2020   
3210  5248477                                               风景园林  2020   
3212  5254834                                              工业工程类  2020   
3234  5325025                                               风景园林  2020   
3252  5382814                                               工业工程  2020   
3288  5499364                                         工业工程（国家专项）  2020   

      minScore  categoryID_id  collegeID_id   c1        c2  
1652       588              2         10759  管理学     工业工程类  
1668       587              2         10759   工学     农业工程类  
1919       654              2         10030   工学      计算机类  
2008       641              2         10022   工学       建筑类  
2092       594              2         10755  管理学     工业工程类  
2165       594              2         10755  管理学     工业工程类  
2307       609              2         10216  管理学     工业工程类  
2575       614              2         10291   工学       建筑类  
2685       633              2         10635   工学       建筑类  
2733       657              2        106111   工学       建筑类  
2742       654              2        106111   工学   生物医学工程类  
2784       655              2         10141   工学       机械类  
2799       654              2         10141   工学       机械类  
2801       652              2         10141   工学       材料类  
2809       653              2         10141   工学       机械类  
2834       609              2         11078   工学       建筑类  
2959       646              2         10183   工学       地质类  
2960       652              2         10183   工学   生物医学工程类  
2970       650              2         10183   工学       机械类  
2972       646              2         10183   工学     交通运输类  
2988       646              2         10183   工学     生物工程类  
2992       653              2         10183   理学  数学类,统计学类  
3011       650              2         10183   工学       机械类  
3013       645              2         10183   工学     交通运输类  
3016       638              2         10183

In [1129]:
df7.to_csv('major_last.csv',encoding='gbk')

In [1127]:
df7

id         majorName    year  minScore  categoryID_id  \
0      183679               波斯语  2020.0     623.0            1.0   
1      183715              阿拉伯语  2020.0     624.0            1.0   
1784  5541770      工程管理(中外合作办学)     NaN       NaN            NaN   
1784  5541770      工程管理(中外合作办学)     NaN       NaN            NaN   
1784  5541770      工程管理(中外合作办学)     NaN       NaN            NaN   
...       ...               ...     ...       ...            ...   
1784  5541770      工程管理(中外合作办学)     NaN       NaN            NaN   
3298  5531969  计算机科学与技术(中外合作办学)  2020.0     589.0            2.0   
3299  5535235        化学(中外合作办学)  2020.0     577.0            2.0   
3300  5538502      生物技术(中外合作办学)  2020.0     575.0            2.0   
1784  5541770      工程管理(中外合作办学)     NaN       NaN            NaN   

      collegeID_id   c1        c2  collage  score  
0          10036.0   文学   外国语言文学类      NaN    NaN  
1          10036.0   文学   外国语言文学类      NaN    NaN  
1784           NaN  管理学  管理科学与工程类  10225.0  547.0  
1784           NaN  管理学  管理科学与工程类  10225.0  547.0  
1784           NaN  管理学  管理科学与工程类  10225.0  547.0  
...            ...  ...       ...      ...    ...  
1784           NaN  管理学  管理科学与工程类  10225.0  547.0  
3298       10225.0   工学      计算机类      NaN    NaN  
3299       10225.0   理学       化学类      NaN    NaN  
3300       10225.0   理学     生物科学类      NaN    NaN  
1784           NaN  管理学  管理科学与工程类  10225.0  547.0  

[3296 rows x 10 columns]

In [1041]:
df6.loc[index,:]

id                                 2044384
majorName    人文科学试验班（含汉语言文学、历史学、哲学）<励耘实验班>
score                                  647
c1                                      文学
c2                                     NaN
collage                              10027
Name: 949, dtype: object

In [1124]:
df6

id              majorName  score   c1          c2  collage
0      183679                    波斯语    623   文学     外国语言文学类    10036
1      183715                   阿拉伯语    624   文学     外国语言文学类    10036
2      183716                    朝鲜语    624   文学     外国语言文学类    10036
3      183753  中国语言文学类(汉语国际教育、汉语言文学)    625   文学     中国语言文学类    10036
4      183754     政治学类(国际政治、政治学与行政学)    625   法学        政治学类    10036
...       ...                    ...    ...  ...         ...      ...
1608  5509135             动物科学（国家专项）    613   农学       动物生产类    10225
1609  5522177             物流工程（国家专项）    613  管理学    物流管理与工程类    10225
1610  5525440               俄语（国家专项）    611   文学     外国语言文学类    10225
1611  5528704     野生动物与自然保护区管理（国家专项）    611   农学  自然保护与环境生态类    10225
1612  5541770           工程管理(中外合作办学)    547  管理学    管理科学与工程类    10225

[1613 rows x 6 columns]

In [1043]:
decide.loc[i+1,:]

id              3249439.0
majorName         社会科学试验班
score               639.0
c1           文学,法学,哲学,历史学
c2                    NaN
collage           10255.0
Name: 26, dtype: object

In [1123]:
for i in range(37):
    index=df6[df6['id']==decide.loc[i,'id']].index[0]
    if df6.loc[index,'c1']==decide.loc[i,'c1']:
        df6.loc[index,'c2']+=','+decide.loc[i,'c2']
        decide.drop(index=i)
    print(i)

0
1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36


In [1122]:
decide#=decide.sort_values(by='id',ignore_index=True)

id                                          majorName  score   c1  \
0    183851.0                              经济学类(国际经济与贸易、财政学、税收学)  628.0  经济学   
1    183851.0                              经济学类(国际经济与贸易、财政学、税收学)  628.0  经济学   
2    184865.0                              电子商务类(电子商务、信息管理与信息系统)  652.0  管理学   
3    185155.0                              经济学类(国际经济与贸易、财政学、税收学)  659.0  经济学   
4    185155.0                              经济学类(国际经济与贸易、财政学、税收学)  659.0  经济学   
5    220727.0                                   法医学（法医学与法学双学士学位）  669.0   医学   
6    245074.0                         林学类（含林学、森林保护、水土保持与荒漠化防治专业）  537.0   农学   
7    335684.0                            公共管理类（包含专业：行政管理、人力资源管理）  605.0  管理学   
8    444874.0         中国语言文学类（包含专业：汉语言文学、汉语国际教育、编辑出版学；办学地点：阳逻校区）  597.0   文学   
9    459064.0  经济学类（4500元/年；包含专业经济学、国际经济与贸易、金融工程、房地产开发与管理、工商管...  641.0  经济学   
10   459064.0  经济学类（4500元/年；包含专业经济学、国际经济与贸易、金融工程、房地产开发与管理、工商管...  641.0  经济学   
11   475135.0  经济学类（包含专业：经济学、国际经济与贸易、工商管理、人力资源管理、财务会计教育、经济学（法...  618.0  经济学   
12   780727.0                                         医学试验班（护理学）  621.0   医学   
13  1240627.0                                                中药学  621.0   医学   
14  1573877.0                                                中药学  586.0   医学   
15  1966484.0                                            新闻学（日语）  610.0   文学   
16  1998580.0                                         政治学、经济学与哲学  639.0   法学   
17  1998580.0                                         政治学、经济学与哲学  639.0   哲学   
18  2042464.0         公共管理类（含信息管理与信息系统<大数据治理方向>、人力资源管理、公共事业管理专业）  639.0  管理学   
19  2042464.0         公共管理类（含信息管理与信息系统<大数据治理方向>、人力资源管理、公共事业管理专业）  639.0  管理学   
20  3870919.0                                     生物医学工程类（生物与药学）  654.0   医学   
21  4502935.0          管理科学与工程类（含信息管理与信息系统、工程管理、物流管理、大数据管理与应用）    649.0  管理学   
22  4511740.0                          会计学（含会计学、财务管理、信用管理、经济学）    649.0  经济学   
23  4523494.0                       经济学类（含经济学、财政学、金融学、国际经济与贸易）    654.0  经济学   
24  4523494.0                       经济学类（含经济学、财政学、金融学、国际经济与贸易）    654.0  经济学   
25  4523494.0                       经济学类（含经济学、财政学、金融学、国际经济与贸易）    654.0  经济学   
26  4627024.0           工商管理类（含工商管理、市场营销、人力资源管理、信息管理与信息系统）  国家专项  645.0  管理学   
27  4630000.0      管理科学与工程类（含信息管理与信息系统、工程管理、物流管理、大数据管理与应用）  国家专项  646.0  管理学   
28  4650860.0                   植物生产类（含农学、园艺、植物保护、农业资源与环境）  国家专项  638.0   农学   
29  4653844.0                                         动物科学  农学单列  617.0   农学   
30  4656829.0                                     动物医学（五年）  农学单列  629.0   农学   
31  4659815.0                   植物生产类（含农学、园艺、植物保护、农业资源与环境）  农学单列  620.0   农学   
32  4671769.0                       经济学类（含经济学、财政学、金融学、国际经济与贸易）    625.0  经济学   
33  4671769.0                       经济学类（含经济学、财政学、金融学、国际经济与贸易）    625.0  经济学   
34  4671769.0                       经济学类（含经济学、财政学、金融学、国际经济与贸易）    625.0  经济学   
35  4994380.0                                  经济学类（经济学、国际经济与贸易）  610.0  经济学   
36  5012977.0                                  经济学类（经济学、国际经济与贸易）  599.0  经济学   

          c2   collage  
0       经济学类   10036.0  
1     经济与贸易类   10036.0  
2   管理科学与工程类   10036.0  
3     经济与贸易类   10036.0  
4       经济学类   10036.0  
5       法医学类   10610.0  
6        林学类   10389.0  
7      公共管理类   10403.0  
8     新闻传播学类   10512.0  
9       经济学类   10511.0  
10    经济与贸易类   10511.0  
11    经济与贸易类   10511.0  
12     临床医学类   10698.0  
13       药学类   10307.0  
14      中药学类   10759.0  
15   外国语言文学类   10030.0  
16      政治学类   10027.0  
17       哲学类   10027.0  
18     工商管理类   10027.0  
19     公共管理类   10027.0  
20     基础医学类  106111.0  
21  管理科学与工程类   10183.0  
22      经济学类   10183.0  
23      经济学类   10183.0  
24      金融学类   10183.0  
25      财政学类   10183.0  
26  管理科学与工程类   10183.0  
27  物流管理与工程类   10183.0  
28     植物生产类   10183.0  
29     植物生产类   10183.0  
30     动物医学类   10183.0  
31     植物生产类   10183.0  
32      财政学类   10183.0  
33    经济与贸易类   10183.0  
34      金融学类   10183.0  
35      经济学类   10184.0  
36    经济与贸易类   10184.0

In [1110]:
df4[1020:1030]

id                                   majorName  score         c1  \
1020  2042464  公共管理类（含信息管理与信息系统<大数据治理方向>、人力资源管理、公共事业管理专业）    639        管理学   
1021  2042464  公共管理类（含信息管理与信息系统<大数据治理方向>、人力资源管理、公共事业管理专业）    639        管理学   
1022  2044384               人文科学试验班（含汉语言文学、历史学、哲学）<励耘实验班>    647  文学,哲学,历史学   
1023  2067502                                       工商管理     617        管理学   
1024  2092709                                        会计学     610        管理学   
1025  2100499                                         法学     594         法学   
1026  2102449                                      汉语言文学     593         文学   
1027  2104400                                       知识产权     593         法学   
1028  2106352                         国际经济与贸易（专业教学背景为英语）     593        经济学   
1029  2108305                               英语（只招英语语种考生）     592         文学   

           c2  collage  
1020    公共管理类    10027  
1021    工商管理类    10027  
1022  中国语言文学类    10027  
1023    工商管理类    10009  
1024    工商管理类    10009  
1025      法学类    10009  
1026  中国语言文学类    10009  
1027      法学类    10009  
1028   经济与贸易类    10009  
1029  外国语言文学类    10009

In [1118]:
# df6:a new df4
df6=df4.loc[[0,1],:]
df6=pd.DataFrame(df6)
decide=pd.DataFrame()
for i in range(2,1785):
    new=df4.loc[i,:]
    print(i)
    #print(new,df6.loc[df6.shape[0]-1,:])
    if new['id']!=df6.loc[df6.shape[0]-1,:]['id']:
        print('yes')
        df6=df6.append(new,ignore_index=True)
    else:
        if new['c1']==df6.loc[df6.shape[0]-1,:]['c1']:
            if new['c2']!=df6.loc[df6.shape[0]-1,:]['c2']:
                decide=decide.append(new)
        else:
            decide=decide.append(new)
df6

2
yes
3
yes
4
yes
5
yes
6
yes
7
yes
8
yes
9
yes
10
yes
11
12
13
yes
14
15
yes
16
yes
17
yes
18
yes
19
yes
20
yes
21
yes
22
yes
23
24
yes
25
yes
26
yes
27
yes
28
29
yes
30
yes
31
yes
32
yes
33
yes
34
yes
35
yes
36
37
38
yes
39
yes
40
yes
41
yes
42
yes
43
yes
44
yes
45
yes
46
yes
47
yes
48
yes
49
yes
50
yes
51
yes
52
yes
53
yes
54
yes
55
56
yes
57
yes
58
yes
59
yes
60
yes
61
yes
62
yes
63
64
yes
65
yes
66
yes
67
yes
68
yes
69
yes
70
yes
71
yes
72
yes
73
yes
74
yes
75
yes
76
yes
77
yes
78
yes
79
yes
80
yes
81
yes
82
yes
83
yes
84
yes
85
yes
86
yes
87
yes
88
89
yes
90
yes
91
yes
92
yes
93
yes
94
yes
95
96
yes
97
yes
98
yes
99
yes
100
yes
101
yes
102
yes
103
yes
104
yes
105
yes
106
yes
107
yes
108
yes
109
yes
110
yes
111
yes
112
yes
113
114
yes
115
yes
116
yes
117
yes
118
yes
119
yes
120
yes
121
yes
122
yes
123
yes
124
yes
125
yes
126
yes
127
yes
128
yes
129
yes
130
yes
131
yes
132
yes
133
yes
134
yes
135
yes
136
137
yes
138
yes
139
yes
140
yes
141
yes
142
yes
143
yes
144
yes
145
yes
146
ye

yes
1079
yes
1080
yes
1081
yes
1082
yes
1083
1084
yes
1085
yes
1086
yes
1087
1088
yes
1089
yes
1090
yes
1091
yes
1092
yes
1093
yes
1094
yes
1095
yes
1096
yes
1097
yes
1098
yes
1099
yes
1100
yes
1101
1102
yes
1103
yes
1104
yes
1105
1106
yes
1107
yes
1108
yes
1109
yes
1110
yes
1111
yes
1112
yes
1113
yes
1114
yes
1115
yes
1116
yes
1117
1118
yes
1119
yes
1120
yes
1121
1122
yes
1123
yes
1124
yes
1125
yes
1126
yes
1127
yes
1128
yes
1129
yes
1130
yes
1131
yes
1132
yes
1133
yes
1134
yes
1135
1136
yes
1137
yes
1138
yes
1139
yes
1140
1141
yes
1142
yes
1143
yes
1144
yes
1145
1146
yes
1147
yes
1148
yes
1149
yes
1150
yes
1151
yes
1152
1153
yes
1154
1155
yes
1156
yes
1157
yes
1158
1159
yes
1160
1161
yes
1162
yes
1163
yes
1164
yes
1165
yes
1166
yes
1167
yes
1168
yes
1169
yes
1170
yes
1171
yes
1172
yes
1173
yes
1174
yes
1175
yes
1176
1177
yes
1178
1179
yes
1180
yes
1181
yes
1182
yes
1183
yes
1184
yes
1185
yes
1186
yes
1187
1188
yes
1189
yes
1190
yes
1191
yes
1192
yes
1193
yes
1194
yes
1195
yes
1196
11

id              majorName  score   c1          c2  collage
0      183679                    波斯语    623   文学     外国语言文学类    10036
1      183715                   阿拉伯语    624   文学     外国语言文学类    10036
2      183716                    朝鲜语    624   文学     外国语言文学类    10036
3      183753  中国语言文学类(汉语国际教育、汉语言文学)    625   文学     中国语言文学类    10036
4      183754     政治学类(国际政治、政治学与行政学)    625   法学        政治学类    10036
...       ...                    ...    ...  ...         ...      ...
1608  5509135             动物科学（国家专项）    613   农学       动物生产类    10225
1609  5522177             物流工程（国家专项）    613  管理学    物流管理与工程类    10225
1610  5525440               俄语（国家专项）    611   文学     外国语言文学类    10225
1611  5528704     野生动物与自然保护区管理（国家专项）    611   农学  自然保护与环境生态类    10225
1612  5541770           工程管理(中外合作办学)    547  管理学    管理科学与工程类    10225

[1613 rows x 6 columns]

In [1099]:
df1[df1['c2'].isnull()]

id              majorName  year  minScore  categoryID_id  \
0      183679                    波斯语  2020       623              1   
1      183715                   阿拉伯语  2020       624              1   
2      183716                    朝鲜语  2020       624              1   
3      183753  中国语言文学类(汉语国际教育、汉语言文学)  2020       625              1   
4      183754     政治学类(国际政治、政治学与行政学)  2020       625              1   
...       ...                    ...   ...       ...            ...   
3291  5509135             动物科学（国家专项）  2020       613              2   
3295  5522177             物流工程（国家专项）  2020       613              2   
3296  5525440               俄语（国家专项）  2020       611              2   
3297  5528704     野生动物与自然保护区管理（国家专项）  2020       611              2   
3301  5541770           工程管理(中外合作办学)  2020       547              2   

      collegeID_id    c1    c2  
0            10036  None  None  
1            10036  None  None  
2            10036  None  None  
3            10036  None  None  
4            10036  None  None  
...            ...   ...   ...  
3291         10225  None  None  
3295         10225  None  None  
3296         10225  None  None  
3297         10225  None  None  
3301         10225  None  None  

[1583 rows x 8 columns]

In [1079]:
df4[df4['majorName'].str.contains('文科试验班类')]

id         majorName  score            c1   c2  collage
641   928045            文科试验班类    631  文学,法学,经济学,哲学  NaN    10286
791  1431889  文科试验班类（经济、金融与管理）    672       经济学,管理学  NaN    10003
792  1433459      文科试验班类（新雅书院）    669     法学,经济学,哲学  NaN    10003
798  1453960            文科试验班类    671  文学,法学,经济学,哲学  NaN    10001

In [1082]:
a=[796]
df4.loc[a,'c1']='文学,哲学,历史学'

In [1070]:
a=df4[df4['majorName']=='社会科学试验班(精品文科班)'].index
a=list(a)
df4.loc[a,'c1']='法学,经济学,管理学,教育学'
df4.loc[a,:]

id       majorName  score              c1   c2  collage
899  1675127  社会科学试验班(精品文科班)    629  法学,经济学,管理学,教育学  NaN    10007

In [1060]:
df4=df4.drop(index=[1022,1609])

In [1162]:
df7=df7.sort_values(by='id',ignore_index=True)

In [1131]:
df4[df4['id']==1458715]

id majorName  score  c1   c2  collage
800  1458715        法学     66  法学  法学类    10001

In [1161]:
df7[df7['id']==1458715]

id majorName  year  minScore  categoryID_id  collegeID_id  c1   c2
1619  1458715        法学  2020       666              1         10001  法学  法学类

In [1160]:
df7.loc[1619,'minScore']=666

In [1164]:
df7.to_csv('major_last.csv',encoding='gbk')

In [1097]:
df6

id                                   majorName  score   c1  \
0     183679                                         波斯语    623   文学   
1     183715                                        阿拉伯语    624   文学   
2     183716                                         朝鲜语    624   文学   
3     183753                       中国语言文学类(汉语国际教育、汉语言文学)    625   文学   
4     183754                          政治学类(国际政治、政治学与行政学)    625   法学   
..       ...                                         ...    ...  ...   
944  2015689                                       汉语言文学    643   文学   
945  2017595                                          英语    668   文学   
946  2019502                                传播学（网络新媒体方向）    639   文学   
947  2021410                                         历史学    640  历史学   
948  2042464  公共管理类（含信息管理与信息系统<大数据治理方向>、人力资源管理、公共事业管理专业）    639  管理学   

                       c2  collage  
0                 外国语言文学类    10036  
1                 外国语言文学类    10036  
2                 外国语言文学类    10036  
3                 中国语言文学类    10036  
4                    政治学类    10036  
..                    ...      ...  
944               中国语言文学类    10027  
945               外国语言文学类    10027  
946                新闻传播学类    10027  
947                  历史学类    10027  
948  管理科学与工程类,公共管理类,工商管理类    10027  

[949 rows x 6 columns]

In [1156]:
No_label=[]
Do_label=[]
df7=df1.loc[[0,1],:]
df7=pd.DataFrame(df7)
df7.loc[[0,1],['c1','c2']]=df6.loc[[0,1],['c1','c2']]

for i in range(2,3302):
    aid=df1.loc[i,'id']
    ind=list(df6[df6['id']==aid].index)
    if len(ind)==0:#未在df6标记中
        if df1.loc[i,'c1']==None:#也不在df1标记中
            No_label.append(i)
        else:#在df1标记中
            df7=df7.append(df1.loc[i,:])
    else:#在df6标记中
        if df1.loc[i,'c1']==None:#不在df1标记中
            #for ind in indexes:#可能有多个标记
             #   new=df1.loc[i,:]
             #   new['c1']=df6.loc[ind,'c1']
             #   if type(df6.loc[ind,'c2'])==str:
            #      new['c2']=df6.loc[ind,'c2']
            new=df1.loc[i,:]
            new['c1']=df6.loc[ind,'c1'].values[0]
            if type(df6.loc[ind,'c2'].values[0])==str:
                new['c2']=df6.loc[ind,'c2'].values[0]
            df7=df7.append(new)
        else:#也在df1标记中,以df1为准
            df7=df7.append(df1.loc[i,:])
            Do_label.append(i)
    
    print(i,'/ 3301  finished!')

E:\anaconda\lib\site-packages\pandas\core\series.py:1056: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  cacher_needs_updating = self._check_is_chained_assignment_possible()


2 / 3301  finished!
3 / 3301  finished!
4 / 3301  finished!
5 / 3301  finished!
6 / 3301  finished!
7 / 3301  finished!
8 / 3301  finished!
9 / 3301  finished!
10 / 3301  finished!
11 / 3301  finished!
12 / 3301  finished!
13 / 3301  finished!
14 / 3301  finished!
15 / 3301  finished!
16 / 3301  finished!
17 / 3301  finished!
18 / 3301  finished!
19 / 3301  finished!
20 / 3301  finished!
21 / 3301  finished!
22 / 3301  finished!
23 / 3301  finished!
24 / 3301  finished!
25 / 3301  finished!
26 / 3301  finished!
27 / 3301  finished!
28 / 3301  finished!
29 / 3301  finished!
30 / 3301  finished!
31 / 3301  finished!
32 / 3301  finished!
33 / 3301  finished!
34 / 3301  finished!
35 / 3301  finished!
36 / 3301  finished!
37 / 3301  finished!
38 / 3301  finished!
39 / 3301  finished!
40 / 3301  finished!
41 / 3301  finished!
42 / 3301  finished!
43 / 3301  finished!
44 / 3301  finished!
45 / 3301  finished!
46 / 3301  finished!
47 / 3301  finished!
48 / 3301  finished!
49 / 3301  finished!


398 / 3301  finished!
399 / 3301  finished!
400 / 3301  finished!
401 / 3301  finished!
402 / 3301  finished!
403 / 3301  finished!
404 / 3301  finished!
405 / 3301  finished!
406 / 3301  finished!
407 / 3301  finished!
408 / 3301  finished!
409 / 3301  finished!
410 / 3301  finished!
411 / 3301  finished!
412 / 3301  finished!
413 / 3301  finished!
414 / 3301  finished!
415 / 3301  finished!
416 / 3301  finished!
417 / 3301  finished!
418 / 3301  finished!
419 / 3301  finished!
420 / 3301  finished!
421 / 3301  finished!
422 / 3301  finished!
423 / 3301  finished!
424 / 3301  finished!
425 / 3301  finished!
426 / 3301  finished!
427 / 3301  finished!
428 / 3301  finished!
429 / 3301  finished!
430 / 3301  finished!
431 / 3301  finished!
432 / 3301  finished!
433 / 3301  finished!
434 / 3301  finished!
435 / 3301  finished!
436 / 3301  finished!
437 / 3301  finished!
438 / 3301  finished!
439 / 3301  finished!
440 / 3301  finished!
441 / 3301  finished!
442 / 3301  finished!
443 / 3301

774 / 3301  finished!
775 / 3301  finished!
776 / 3301  finished!
777 / 3301  finished!
778 / 3301  finished!
779 / 3301  finished!
780 / 3301  finished!
781 / 3301  finished!
782 / 3301  finished!
783 / 3301  finished!
784 / 3301  finished!
785 / 3301  finished!
786 / 3301  finished!
787 / 3301  finished!
788 / 3301  finished!
789 / 3301  finished!
790 / 3301  finished!
791 / 3301  finished!
792 / 3301  finished!
793 / 3301  finished!
794 / 3301  finished!
795 / 3301  finished!
796 / 3301  finished!
797 / 3301  finished!
798 / 3301  finished!
799 / 3301  finished!
800 / 3301  finished!
801 / 3301  finished!
802 / 3301  finished!
803 / 3301  finished!
804 / 3301  finished!
805 / 3301  finished!
806 / 3301  finished!
807 / 3301  finished!
808 / 3301  finished!
809 / 3301  finished!
810 / 3301  finished!
811 / 3301  finished!
812 / 3301  finished!
813 / 3301  finished!
814 / 3301  finished!
815 / 3301  finished!
816 / 3301  finished!
817 / 3301  finished!
818 / 3301  finished!
819 / 3301

1150 / 3301  finished!
1151 / 3301  finished!
1152 / 3301  finished!
1153 / 3301  finished!
1154 / 3301  finished!
1155 / 3301  finished!
1156 / 3301  finished!
1157 / 3301  finished!
1158 / 3301  finished!
1159 / 3301  finished!
1160 / 3301  finished!
1161 / 3301  finished!
1162 / 3301  finished!
1163 / 3301  finished!
1164 / 3301  finished!
1165 / 3301  finished!
1166 / 3301  finished!
1167 / 3301  finished!
1168 / 3301  finished!
1169 / 3301  finished!
1170 / 3301  finished!
1171 / 3301  finished!
1172 / 3301  finished!
1173 / 3301  finished!
1174 / 3301  finished!
1175 / 3301  finished!
1176 / 3301  finished!
1177 / 3301  finished!
1178 / 3301  finished!
1179 / 3301  finished!
1180 / 3301  finished!
1181 / 3301  finished!
1182 / 3301  finished!
1183 / 3301  finished!
1184 / 3301  finished!
1185 / 3301  finished!
1186 / 3301  finished!
1187 / 3301  finished!
1188 / 3301  finished!
1189 / 3301  finished!
1190 / 3301  finished!
1191 / 3301  finished!
1192 / 3301  finished!
1193 / 3301

1508 / 3301  finished!
1509 / 3301  finished!
1510 / 3301  finished!
1511 / 3301  finished!
1512 / 3301  finished!
1513 / 3301  finished!
1514 / 3301  finished!
1515 / 3301  finished!
1516 / 3301  finished!
1517 / 3301  finished!
1518 / 3301  finished!
1519 / 3301  finished!
1520 / 3301  finished!
1521 / 3301  finished!
1522 / 3301  finished!
1523 / 3301  finished!
1524 / 3301  finished!
1525 / 3301  finished!
1526 / 3301  finished!
1527 / 3301  finished!
1528 / 3301  finished!
1529 / 3301  finished!
1530 / 3301  finished!
1531 / 3301  finished!
1532 / 3301  finished!
1533 / 3301  finished!
1534 / 3301  finished!
1535 / 3301  finished!
1536 / 3301  finished!
1537 / 3301  finished!
1538 / 3301  finished!
1539 / 3301  finished!
1540 / 3301  finished!
1541 / 3301  finished!
1542 / 3301  finished!
1543 / 3301  finished!
1544 / 3301  finished!
1545 / 3301  finished!
1546 / 3301  finished!
1547 / 3301  finished!
1548 / 3301  finished!
1549 / 3301  finished!
1550 / 3301  finished!
1551 / 3301

1885 / 3301  finished!
1886 / 3301  finished!
1887 / 3301  finished!
1888 / 3301  finished!
1889 / 3301  finished!
1890 / 3301  finished!
1891 / 3301  finished!
1892 / 3301  finished!
1893 / 3301  finished!
1894 / 3301  finished!
1895 / 3301  finished!
1896 / 3301  finished!
1897 / 3301  finished!
1898 / 3301  finished!
1899 / 3301  finished!
1900 / 3301  finished!
1901 / 3301  finished!
1902 / 3301  finished!
1903 / 3301  finished!
1904 / 3301  finished!
1905 / 3301  finished!
1906 / 3301  finished!
1907 / 3301  finished!
1908 / 3301  finished!
1909 / 3301  finished!
1910 / 3301  finished!
1911 / 3301  finished!
1912 / 3301  finished!
1913 / 3301  finished!
1914 / 3301  finished!
1915 / 3301  finished!
1916 / 3301  finished!
1917 / 3301  finished!
1918 / 3301  finished!
1919 / 3301  finished!
1920 / 3301  finished!
1921 / 3301  finished!
1922 / 3301  finished!
1923 / 3301  finished!
1924 / 3301  finished!
1925 / 3301  finished!
1926 / 3301  finished!
1927 / 3301  finished!
1928 / 3301

2254 / 3301  finished!
2255 / 3301  finished!
2256 / 3301  finished!
2257 / 3301  finished!
2258 / 3301  finished!
2259 / 3301  finished!
2260 / 3301  finished!
2261 / 3301  finished!
2262 / 3301  finished!
2263 / 3301  finished!
2264 / 3301  finished!
2265 / 3301  finished!
2266 / 3301  finished!
2267 / 3301  finished!
2268 / 3301  finished!
2269 / 3301  finished!
2270 / 3301  finished!
2271 / 3301  finished!
2272 / 3301  finished!
2273 / 3301  finished!
2274 / 3301  finished!
2275 / 3301  finished!
2276 / 3301  finished!
2277 / 3301  finished!
2278 / 3301  finished!
2279 / 3301  finished!
2280 / 3301  finished!
2281 / 3301  finished!
2282 / 3301  finished!
2283 / 3301  finished!
2284 / 3301  finished!
2285 / 3301  finished!
2286 / 3301  finished!
2287 / 3301  finished!
2288 / 3301  finished!
2289 / 3301  finished!
2290 / 3301  finished!
2291 / 3301  finished!
2292 / 3301  finished!
2293 / 3301  finished!
2294 / 3301  finished!
2295 / 3301  finished!
2296 / 3301  finished!
2297 / 3301

2617 / 3301  finished!
2618 / 3301  finished!
2619 / 3301  finished!
2620 / 3301  finished!
2621 / 3301  finished!
2622 / 3301  finished!
2623 / 3301  finished!
2624 / 3301  finished!
2625 / 3301  finished!
2626 / 3301  finished!
2627 / 3301  finished!
2628 / 3301  finished!
2629 / 3301  finished!
2630 / 3301  finished!
2631 / 3301  finished!
2632 / 3301  finished!
2633 / 3301  finished!
2634 / 3301  finished!
2635 / 3301  finished!
2636 / 3301  finished!
2637 / 3301  finished!
2638 / 3301  finished!
2639 / 3301  finished!
2640 / 3301  finished!
2641 / 3301  finished!
2642 / 3301  finished!
2643 / 3301  finished!
2644 / 3301  finished!
2645 / 3301  finished!
2646 / 3301  finished!
2647 / 3301  finished!
2648 / 3301  finished!
2649 / 3301  finished!
2650 / 3301  finished!
2651 / 3301  finished!
2652 / 3301  finished!
2653 / 3301  finished!
2654 / 3301  finished!
2655 / 3301  finished!
2656 / 3301  finished!
2657 / 3301  finished!
2658 / 3301  finished!
2659 / 3301  finished!
2660 / 3301

2980 / 3301  finished!
2981 / 3301  finished!
2982 / 3301  finished!
2983 / 3301  finished!
2984 / 3301  finished!
2985 / 3301  finished!
2986 / 3301  finished!
2987 / 3301  finished!
2988 / 3301  finished!
2989 / 3301  finished!
2990 / 3301  finished!
2991 / 3301  finished!
2992 / 3301  finished!
2993 / 3301  finished!
2994 / 3301  finished!
2995 / 3301  finished!
2996 / 3301  finished!
2997 / 3301  finished!
2998 / 3301  finished!
2999 / 3301  finished!
3000 / 3301  finished!
3001 / 3301  finished!
3002 / 3301  finished!
3003 / 3301  finished!
3004 / 3301  finished!
3005 / 3301  finished!
3006 / 3301  finished!
3007 / 3301  finished!
3008 / 3301  finished!
3009 / 3301  finished!
3010 / 3301  finished!
3011 / 3301  finished!
3012 / 3301  finished!
3013 / 3301  finished!
3014 / 3301  finished!
3015 / 3301  finished!
3016 / 3301  finished!
3017 / 3301  finished!
3018 / 3301  finished!
3019 / 3301  finished!
3020 / 3301  finished!
3021 / 3301  finished!
3022 / 3301  finished!
3023 / 3301

In [1154]:
df6.loc[ind,'c2']

1612    管理科学与工程类
Name: c2, dtype: object

In [1155]:
type(df6.loc[ind,'c2'].values[0])==str

True

In [1163]:
df7

id              majorName  year  minScore  categoryID_id  \
0      183679                    波斯语  2020       623              1   
1      183715                   阿拉伯语  2020       624              1   
2      183716                    朝鲜语  2020       624              1   
3      183753  中国语言文学类(汉语国际教育、汉语言文学)  2020       625              1   
4      183754     政治学类(国际政治、政治学与行政学)  2020       625              1   
...       ...                    ...   ...       ...            ...   
3291  5528704     野生动物与自然保护区管理（国家专项）  2020       611              2   
3292  5531969       计算机科学与技术(中外合作办学)  2020       589              2   
3293  5535235             化学(中外合作办学)  2020       577              2   
3294  5538502           生物技术(中外合作办学)  2020       575              2   
3295  5541770           工程管理(中外合作办学)  2020       547              2   

      collegeID_id   c1          c2  
0            10036   文学     外国语言文学类  
1            10036   文学     外国语言文学类  
2            10036   文学     外国语言文学类  
3            10036   文学     中国语言文学类  
4            10036   法学        政治学类  
...            ...  ...         ...  
3291         10225   农学  自然保护与环境生态类  
3292         10225   工学        计算机类  
3293         10225   理学         化学类  
3294         10225   理学       生物科学类  
3295         10225  管理学    管理科学与工程类  

[3296 rows x 8 columns]

In [1126]:
df1.loc[No_label,:]

id majorName  year  minScore  categoryID_id  collegeID_id    c1    c2
106  202925       all  2020       614              1         10559  None  None
107  202999       all  2020       636              2         10559  None  None
239  221545       all  2020       618              2         10593  None  None
240  221752       all  2020       600              1         10593  None  None
313  239564      高校专项  2020       571              2         10542  None  None
322  242129      高校专项  2020       568              1         10542  None  None

In [1104]:
idlist=list(df1.loc[No_label,:]['id'].values)
dl_index=[]
for sid in idlist:
    dl_index.extend(list(df4[df4['id']==sid].index))
dl_index

[1023,
 1024,
 1025,
 1026,
 1027,
 1028,
 1029,
 1030,
 1031,
 1032,
 1033,
 1034,
 1035,
 1036,
 1037,
 1038,
 1039,
 1040,
 1041,
 1042,
 1043,
 1045,
 1046,
 1047,
 1048,
 1049,
 1050,
 1051,
 1052,
 1053,
 1054,
 1055,
 1056,
 1057,
 1058,
 1059,
 1060,
 1061,
 1062,
 1063,
 1064,
 1065,
 1066,
 1067,
 1068,
 1069,
 1070,
 1071,
 1072,
 1073,
 1074,
 1075,
 1076,
 1077,
 1078,
 1079,
 1080,
 1081,
 1082,
 1083,
 1084,
 1085,
 1086,
 1087,
 1088,
 1089,
 1090,
 1091,
 1093,
 1094,
 1095,
 1096,
 1097,
 1098,
 1099,
 1100,
 1101,
 1102,
 1103,
 1104,
 1105,
 1106,
 1107,
 1108,
 1109,
 1110,
 1111,
 1112,
 1113,
 1114,
 1115,
 1116,
 1117,
 1118,
 1119,
 1120,
 1121,
 1122,
 1123,
 1124,
 1125,
 1127,
 1128,
 1129,
 1130,
 1131,
 1132,
 1133,
 1134,
 1135,
 1136,
 1137,
 1138,
 1139,
 1140,
 1141,
 1142,
 1143,
 1144,
 1145,
 1146,
 1147,
 1148,
 1149,
 1150,
 1151,
 1152,
 1153,
 1154,
 1155,
 1156,
 1157,
 1158,
 1159,
 1160,
 1161,
 1162,
 1163,
 1164,
 1165,
 1166,
 1167,
 1168,